--- a/package/boot/uboot-rockchip/Makefile
+++ b/package/boot/uboot-rockchip/Makefile
@@ -202,7 +202,28 @@
   BUILD_DEVICES:= \
     friendlyarm_nanopi-r5s
 endef
+
+define U-Boot/opc-h66k-rk3568
+  $(U-Boot/Default/rk3568)
+  NAME:=OPC-H66K
+  BUILD_DEVICES:= \
+    hinlink_opc-h66k
+endef
 
+define U-Boot/opc-h68k-rk3568
+  $(U-Boot/Default/rk3568)
+  NAME:=OPC-H68K
+  BUILD_DEVICES:= \
+    hinlink_opc-h68k
+endef
+
+define U-Boot/opc-h69k-rk3568
+  $(U-Boot/Default/rk3568)
+  NAME:=OPC-H69K
+  BUILD_DEVICES:= \
+    hinlink_opc-h69k
+endef
+
 define U-Boot/photonicat-rk3568
   $(U-Boot/Default/rk3568)
   NAME:=Photonicat
@@ -335,6 +356,9 @@
   mrkaio-m68s-rk3568 \
   nanopi-r5c-rk3568 \
   nanopi-r5s-rk3568 \
+  opc-h66k-rk3568 \
+  opc-h68k-rk3568 \
+  opc-h69k-rk3568 \
   photonicat-rk3568 \
   radxa-e25-rk3568 \
   roc-pc-rk3568 \

--- a/package/boot/uboot-rockchip/patches/900-arm-add-dts-files.patch
+++ b/package/boot/uboot-rockchip/patches/900-arm-add-dts-files.patch
@@ -1,10 +1,12 @@
 --- a/arch/arm/dts/Makefile
 +++ b/arch/arm/dts/Makefile
-@@ -148,6 +148,7 @@ dtb-$(CONFIG_ROCKCHIP_RK3399) += \
+@@ -148,6 +148,9 @@ dtb-$(CONFIG_ROCKCHIP_RK3399) += \
  	rk3399-firefly.dtb \
  	rk3399-gru-bob.dtb \
  	rk3399-gru-kevin.dtb \
 +	rk3399-guangmiao-g4c.dtb \
++	rk3399-h3399pc.dtb \
++	rk3399-mpc1903.dtb \
  	rk3399-khadas-edge.dtb \
  	rk3399-khadas-edge-captain.dtb \
  	rk3399-khadas-edge-v.dtb \
@@ -16,12 +18,17 @@
  	rk3399-orangepi.dtb \
  	rk3399-pinebook-pro.dtb \
  	rk3399-pinephone-pro.dtb \
-@@ -182,12 +184,17 @@ dtb-$(CONFIG_ROCKCHIP_RK3568) += \
+@@ -182,12 +184,22 @@ dtb-$(CONFIG_ROCKCHIP_RK3568) += \
  	rk3566-soquartz-model-a.dtb \
  	rk3568-bpi-r2-pro.dtb \
  	rk3568-evb.dtb \
 +	rk3568-fastrhino-r66s.dtb \
 +	rk3568-fastrhino-r68s.dtb \
++	rk3568-lyt-t68m.dtb \
++	rk3568-xgp.dtb \
++	rk3568-opc-h66k.dtb \
++	rk3568-opc-h68k.dtb \
++	rk3568-opc-h69k.dtb \
  	rk3568-generic.dtb \
  	rk3568-lubancat-2.dtb \
 +	rk3568-mrkaio-m68s.dtb \

--- a/target/linux/rockchip/armv8/base-files/etc/board.d/02_network
+++ b/target/linux/rockchip/armv8/base-files/etc/board.d/02_network
@@ -17,6 +17,7 @@
 	friendlyarm,nanopi-r4se|\
 	friendlyarm,nanopi-r4s-enterprise|\
 	friendlyarm,nanopi-r6c|\
+	hinlink,opc-h66k|\
 	radxa,rockpi-e|\
 	xunlong,orangepi-r1-plus|\
 	xunlong,orangepi-r1-plus-lts)
@@ -34,7 +35,8 @@
 	friendlyarm,nanopi-r5s)
 		ucidef_set_interfaces_lan_wan 'eth1 eth2' 'eth0'
 		;;
-	friendlyarm,nanopi-r6s)
+	friendlyarm,nanopi-r6s|\
+	hinlink,opc-h69k)
 		ucidef_set_interfaces_lan_wan 'eth0 eth1' 'eth2'
 		;;
 	hinlink,opc-h68k|\
@@ -67,7 +69,10 @@
 	friendlyarm,nanopi-r4s|\
 	friendlyarm,nanopi-r5s|\
 	friendlyarm,nanopi-r6s|\
-	sinovoip,rk3568-bpi-r2pro)
+	sinovoip,rk3568-bpi-r2pro|\
+	hinlink,opc-h66k|\
+	hinlink,opc-h68k|\
+	hinlink,opc-h69k)
 		wan_mac=$(macaddr_generate_from_mmc_cid mmcblk1)
 		lan_mac=$(macaddr_add "$wan_mac" 1)
 		;;

--- a/target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity
+++ b/target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity
@@ -34,6 +34,7 @@
 firefly,rk3568-roc-pc|\
 friendlyarm,nanopi-r5c|\
 friendlyarm,nanopi-r6c|\
+hinlink,opc-h66k|\
 lunzn,fastrhino-r66s|\
 radxa,e25|\
 xunlong,orangepi-5-plus)
@@ -60,15 +61,17 @@
 	set_interface_core 20 "eth1"
 	;;
 friendlyarm,nanopi-r5s|\
-friendlyarm,nanopi-r6s)
+friendlyarm,nanopi-r6s|\
+hinlink,opc-h69k)
 	set_interface_core 2 "eth0"
 	set_interface_core 4 "eth1"
 	set_interface_core 8 "eth2"
 	;;
-lunzn,fastrhino-r68s)
+lunzn,fastrhino-r68s|\
+hinlink,opc-h68k)
 	set_interface_core 1 "eth0"
 	set_interface_core 2 "eth1"
 	set_interface_core 4 "eth2"
 	set_interface_core 8 "eth3"
 	;;
 esac

--- a/target/linux/rockchip/patches-6.6/900-arm64-boot-add-dts-files.patch
+++ b/target/linux/rockchip/patches-6.6/900-arm64-boot-add-dts-files.patch
@@ -16,7 +16,7 @@
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-nanopi-r4s-enterprise.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-orangepi.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-pinebook-pro.dtb
-@@ -90,9 +92,11 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-bp
+@@ -90,9 +92,14 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-bp
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-evb1-v10.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-fastrhino-r66s.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-fastrhino-r68s.dtb
@@ -24,6 +24,9 @@
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-lubancat-2.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-nanopi-r5c.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-nanopi-r5s.dtb
++dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-opc-h66k.dtb
++dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-opc-h68k.dtb
++dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-opc-h69k.dtb
 +dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-photonicat.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-odroid-m1.dtb
  dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-radxa-e25.dtb
