--- a/package/boot/uboot-rk35xx/Makefile
+++ b/package/boot/uboot-rk35xx/Makefile
@@ -14,8 +14,8 @@ PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL:=https://github.com/radxa/u-boot
 
 PKG_SOURCE_DATE:=2024-03-01
-PKG_SOURCE_VERSION:=3b2035860d51a75fc70941acf9ae934140991b21
-PKG_MIRROR_HASH:=4cba13e74fff830f946ee52d55c6a94e2de691d45db6cf69c481f91a8063e23f
+PKG_SOURCE_VERSION:=00ac933690e0bf6b2adc2ab507e50bff05327729
+PKG_MIRROR_HASH:=
 
 include $(INCLUDE_DIR)/u-boot.mk
 include $(INCLUDE_DIR)/package.mk
