--- a/package/openwrt/files/etc/init.d/smartdns
+++ b/package/openwrt/files/etc/init.d/smartdns
@@ -381,6 +381,13 @@ start_service()
 {
 	config_load "smartdns"
 	config_foreach load_service "smartdns"
+	uci set shadowsocksr.@global[0].pdnsd_enable='0'
+	uci del shadowsocksr.@global[0].tunnel_forward
+	uci commit shadowsocksr
+	uci set passwall.@global[0].dns_mode='nonuse'
+	uci del passwall.@global[0].up_trust_pdnsd_dns
+	uci del passwall.@global[0].dns_forward
+	uci commit passwall
 }
 
 reload_service()
