From aa9c6e61e4f9ebdd758309662bba315aa6b8d880 Mon Sep 17 00:00:00 2001
From: garypang13 <48883331+garypang13@users.noreply.github.com>
Date: Tue, 15 Sep 2020 18:37:41 +0800
Subject: [PATCH] Update HttpRequest.cc

---
 src/HttpRequest.cc | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/HttpRequest.cc b/src/HttpRequest.cc
index b5fb4f863..30fd15d68 100644
--- a/src/HttpRequest.cc
+++ b/src/HttpRequest.cc
@@ -262,7 +262,6 @@ std::string HttpRequest::createRequest()
     }
     if (!wantDigest.empty()) {
       wantDigest.erase(wantDigest.size() - 2);
-      builtinHds.emplace_back("Want-Digest:", wantDigest);
     }
   }
   for (const auto& builtinHd : builtinHds) {
