--- /feeds/packages/net/aria2/files/aria2.init
+++ /feeds/packages/net/aria2/files/aria2.init
@@ -312,3 +312,6 @@
 
-	sed '/^$/d' "$config_file_tmp" >"$config_file"
+	cd /usr/share/aria2 && sh ./tracker.sh
+	cat /usr/share/aria2/aria2.conf > "$config_file"
+	echo '' >> "$config_file"
+	sed '/^$/d' "$config_file_tmp" >> "$config_file"
 	rm -f "$config_file_tmp"

--- /feeds/luci/applications/luci-app-aria2/luasrc/controller/aria2.lua
+++ /feeds/luci/applications/luci-app-aria2/luasrc/controller/aria2.lua
@@ -16,22 +16,24 @@
 
-	entry({"admin", "services", "aria2"},
+	entry({"admin", "nas"}, firstchild(), "NAS", 45).dependent = false
+
+	entry({"admin", "nas", "aria2"},
 		firstchild(), _("Aria2")).dependent = false
 
-	entry({"admin", "services", "aria2", "config"},
+	entry({"admin", "nas", "aria2", "config"},
 		cbi("aria2/config"), _("Configuration"), 1)
 
-	entry({"admin", "services", "aria2", "file"},
+	entry({"admin", "nas", "aria2", "file"},
 		form("aria2/files"), _("Files"), 2)
 
-	entry({"admin", "services", "aria2", "log"},
+	entry({"admin", "nas", "aria2", "log"},
 		firstchild(), _("Log"), 3)
 
-	entry({"admin", "services", "aria2", "log", "view"},
+	entry({"admin", "nas", "aria2", "log", "view"},
 		template("aria2/log_template"))
 
-	entry({"admin", "services", "aria2", "log", "read"},
+	entry({"admin", "nas", "aria2", "log", "read"},
 		call("action_log_read"))
 
-	entry({"admin", "services", "aria2", "status"},
+	entry({"admin", "nas", "aria2", "status"},
 		call("action_status"))
 
