--- /feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
+++ /feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
@@ -185,2 +185,7 @@
 			_('Additional Hosts files')).optional = true;
+			
+		o = s.taboption("advanced", form.Flag, "filter_aaaa",
+			_('Disable IPv6 DNS forwards'),
+			_('Filter IPv6(AAAA) DNS Query Name Resolve'));
+		o.optional = true;
 
