--- /feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
+++ /feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
@@ -343,2 +343,9 @@
 
+		o = s.taboption("advanced", form.Value, "mini_ttl",
+			_('Minimum TTL to send to clients'),
+			_('Modify DNS entries minimum TTL (max is 86400, 0 is no modify)'))
+		o.optional = true
+		o.datatype = "range(0,86400)"
+		o.placeholder = 0
+
 		s.taboption('tftp', form.Flag, 'enable_tftp',

--- /package/network/services/dnsmasq/files/dnsmasq.init
+++ /package/network/services/dnsmasq/files/dnsmasq.init
@@ -898,2 +898,3 @@
 	append_parm "$cfg" "local_ttl" "--local-ttl"
+	append_parm "$cfg" mini_ttl "--min-ttl"
 	append_parm "$cfg" "max_ttl" "--max-ttl"
