--- /feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
+++ /feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
@@ -343,2 +343,9 @@
 
+		o = s.taboption("advanced", form.Value, "mini_ttl",
+		_("Minimum TTL to send to clients"),
+		_("Modify DNS entries minimum TTL (max is 3600, 0 is no modify)"))
+		o.optional = true
+		o.datatype = "range(0,3600)"
+		o.placeholder = 0
+
 		s.taboption('tftp', form.Flag, 'enable_tftp',
