--- a/target/linux/meson/image/Makefile
+++ b/target/linux/meson/image/Makefile
@@ -112,7 +112,8 @@ endif
 
 define Device/thunder-onecloud
   DEVICE_DTS := meson8b-onecloud
-  DEVICE_TITLE := Thunder OneCloud
+  DEVICE_TITLE := OneCloud 玩客云
+  DEVICE_PACKAGES += kmod-usb-net-rtl8152
   KERNEL_LOADADDR := 0x00208000
 endef
 ifeq ($(SUBTARGET),meson8b)

--- a/package/kernel/linux/modules/netsupport.mk
+++ b/package/kernel/linux/modules/netsupport.mk
@@ -93,7 +93,8 @@ define KernelPackage/vxlan
 	+IPV6:kmod-udptunnel6
   KCONFIG:=CONFIG_VXLAN
   FILES:= \
-	$(LINUX_DIR)/drivers/net/vxlan/vxlan.ko
+	$(LINUX_DIR)/drivers/net/vxlan.ko@lt5.5 \
+	$(LINUX_DIR)/drivers/net/vxlan/vxlan.ko@ge5.6
   AUTOLOAD:=$(call AutoLoad,13,vxlan)
 endef
 