--- a/target/linux/meson/image/Makefile
+++ b/target/linux/meson/image/Makefile
@@ -112,7 +112,8 @@ endif
 
 define Device/thunder-onecloud
   DEVICE_DTS := meson8b-onecloud
-  DEVICE_TITLE := Thunder OneCloud
+  DEVICE_TITLE := OneCloud 玩客云
+  DEVICE_PACKAGES += kmod-usb-net-rtl8152
   KERNEL_LOADADDR := 0x00208000
 endef
 ifeq ($(SUBTARGET),meson8b)

--- a/target/linux/meson/files/arch/arm/boot/dts/meson8b-onecloud.dts
+++ b/target/linux/meson/files/arch/arm/boot/dts/meson8b-onecloud.dts
@@ -137,7 +137,7 @@
 	pinctrl-names = "default";
 
 	phy-handle = <&eth_phy>;
-	phy-mode = "rgmii-id";
+	phy-mode = "rgmii-rxid";
 
 	mdio {
 		compatible = "snps,dwmac-mdio";