--- a/include/target.mk
+++ b/include/target.mk
@@ -44,9 +44,11 @@
 # For router targets
 DEFAULT_PACKAGES.router:=\
-	dnsmasq \
+	dnsmasq-full \
 	firewall \
 	ip6tables \
 	iptables \
 	kmod-ipt-offload \
+	kmod-ipt-nat \
+	kmod-ipt-nat6 \
 	odhcp6c \
 	odhcpd-ipv6only \
@@ -83,4 +85,16 @@
     -include ./$(SUBTARGET)/target.mk
   endif
+endif
+
+ifneq ($(CONFIG_SMALL_FLASH),)
+  DEFAULT_PACKAGES+=-coremark -htop -bash -openssh-sftp-server
+endif
+
+ifeq ($(ARCH),arm)
+  DEFAULT_PACKAGES+=luci-app-cpufreq
+endif
+
+ifeq ($(ARCH),aarch64)
+  DEFAULT_PACKAGES+=luci-app-cpufreq
 endif
 
