--- a/target/linux/mediatek/filogic/base-files/etc/board.d/02_network
+++ b/target/linux/mediatek/filogic/base-files/etc/board.d/02_network
@@ -10,12 +10,17 @@ mediatek_setup_interfaces()
 	case $board in
 	abt,asr3000|\
 	buffalo,wsr-6000ax8|\
-	cmcc,rax3000m|\
+	cmcc,mr3000d-ciq-256m|\
+	cmcc,rax3000m*|\
+	cmcc,xr30*|\
+	imou,lc-hx3001|\
+	umi,uax3000e|\
 	h3c,magic-nx30-pro|\
 	imou,hx21|\
 	konka,komi-a31|\
 	netis,nx30v2|\
 	netis,nx31|\
+	newland,nl-wr8103|\
 	nokia,ea0326gmp|\
 	mercusys,mr80x-v3|\
 	routerich,ax3000-v1|\
@@ -38,6 +43,9 @@ mediatek_setup_interfaces()
 	acer,vero-w6m)
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" internet
 		;;
+	aigo,ags21)
+		ucidef_set_interfaces_lan_wan "lan1 lan2" eth1
+		;;
 	airtel,aap4221zy)
 		ucidef_set_interfaces_lan_wan "lan1 lan2" wan
 		;;
@@ -46,7 +46,10 @@ mediatek_setup_interfaces()
 	asus,zenwifi-bt8|\
 	asus,zenwifi-bt8-ubootmod|\
 	cetron,ct3003|\
-	cmcc,a10-stock|\
+	cmcc,a10|\
+	ikuai,q3000|\
+	philips,hy3000|\
+	sl,3000*|\
 	cmcc,a10-ubootmod|\
 	confiabits,mt7981|\
 	creatlentem,clt-r30b1|\
@@ -70,6 +82,7 @@ mediatek_setup_interfaces()
 	mediatek,mt7981-rfb|\
 	netcore,n60|\
 	netcore,n60-pro|\
+	nradio,c8-660|\
 	nradio,c8-668gl|\
 	ruijie,rg-x60-pro|\
 	unielec,u7981-01*|\
@@ -130,7 +143,8 @@ mediatek_setup_interfaces()
 	cudy,m3000-v1|\
 	cudy,tr3000-256mb-v1|\
 	cudy,tr3000-v1|\
-	cudy,tr3000-v1-ubootmod|\
+	cudy,tr3000-mod|\
+	glinet,gl-mt3600be|\
 	glinet,gl-mt2500|\
 	glinet,gl-mt2500-airoha|\
 	glinet,gl-mt3000|\
@@ -170,6 +186,10 @@ mediatek_setup_interfaces()
 		ucidef_add_switch "switch0" "0:lan0" "1:lan1" "2:lan2" "6@eth0"
 		ucidef_set_interfaces_lan_wan "eth0.1 eth0.2 eth0.3" eth1
 		;;
+	ikuai,q3000)
+		wan_mac=$(mtd_get_mac_binary $part_name 0x10048)
+		lan_mac=$(macaddr_add $wan_mac 1)
+		;;
 	mercusys,mr90x-v1|\
 	mercusys,mr90x-v1-ubi)
 		ucidef_set_interfaces_lan_wan "lan0 lan1 lan2" eth1
@@ -193,11 +213,21 @@ mediatek_setup_interfaces()
 	tplink,re6000xd)
 		ucidef_set_interface_lan "lan1 lan2 eth1"
 		;;
+	philips,hy3000)
+		lan_mac=$(mmc_get_mac_binary factory 0x74)
+		wan_mac=$(macaddr factory $lan_mac 1)
+		label_mac=$wan_mac
+		;;
+ 	sl,3000-emmc)
+ 		lan_mac=$(mmc_get_mac_binary factory 0x04)		
+ 		wan_mac=$(macaddr_add "$lan_mac" -2)
+ 		label_mac=$lan_mac
+ 		;;
 	xiaomi,mi-router-ax3000t|\
 	xiaomi,mi-router-ax3000t-ubootmod|\
-	xiaomi,mi-router-wr30u-stock|\
+	xiaomi,mi-router-wr30u|\
 	xiaomi,mi-router-wr30u-ubootmod|\
-	xiaomi,redmi-router-ax6000-stock|\
+	xiaomi,redmi-router-ax6000|\
 	xiaomi,redmi-router-ax6000-ubootmod)
 		ucidef_set_interfaces_lan_wan "lan2 lan3 lan4" wan
 		;;
@@ -238,6 +268,12 @@ mediatek_setup_macs()
 	bananapi,bpi-r4-lite)
 		wan_mac=$(macaddr_add $(cat /sys/class/net/eth0/address) 1)
 		;;
+	imou,lc-hx3001)
+		lan_mac=$(mtd_get_mac_ascii u-boot-env mac)
+		[ -n "$lan_mac" ] || lan_mac=$(macaddr_add $(mtd_get_mac_binary Factory 0x4) -1)
+		wan_mac=$(macaddr_add $lan_mac 2)
+		label_mac="$lan_mac"
+		;;
 	h3c,magic-nx30-pro)
 		wan_mac=$(mtd_get_mac_ascii pdt_data_1 ethaddr)
 		lan_mac=$(macaddr_add "$wan_mac" 1)
@@ -261,11 +297,17 @@ mediatek_setup_macs()
 		lan_mac=$(mtd_get_mac_ascii u-boot-env mac)
 		label_mac=$lan_mac
 		;;
+	nradio,c8-660|\
 	nradio,c8-668gl)
 		lan_mac=$(mmc_get_mac_ascii bdinfo "fac_mac ")
 		wan_mac=$(macaddr_add "$lan_mac" 2)
 		label_mac=$lan_mac
 		;;
+	newland,nl-wr8103)
+ 		label_mac=$(mtd_get_mac_binary Factory 0x04)		
+ 		wan_mac=$(macaddr_add "$label_mac" -2)
+ 		lan_mac=$label_mac
+ 		;;
 	qihoo,360t7)
 		lan_mac=$(mtd_get_mac_ascii factory lanMac)
 		wan_mac=$(macaddr_add "$lan_mac" 1)
@@ -286,9 +328,9 @@ mediatek_setup_macs()
 		;;
 	xiaomi,mi-router-ax3000t|\
 	xiaomi,mi-router-ax3000t-ubootmod|\
-	xiaomi,mi-router-wr30u-stock|\
+	xiaomi,mi-router-wr30u|\
 	xiaomi,mi-router-wr30u-ubootmod|\
-	xiaomi,redmi-router-ax6000-stock|\
+	xiaomi,redmi-router-ax6000|\
 	xiaomi,redmi-router-ax6000-ubootmod)
 		wan_mac=$(mtd_get_mac_ascii Bdata ethaddr_wan)
 		label_mac=$wan_mac

--- a/target/linux/mediatek/filogic/base-files/etc/hotplug.d/firmware/11-mt76-caldata
+++ b/target/linux/mediatek/filogic/base-files/etc/hotplug.d/firmware/11-mt76-caldata
@@ -17,6 +17,12 @@ case "$FIRMWARE" in
 	tplink,fr365-v1)
 		ln -sf /tmp/wlan/radio /lib/firmware/$FIRMWARE
 		;;
+	umi,uax3000e|\
+	sl,3000-emmc|\
+	cmcc,rax3000m-emmc|\
+	cmcc,xr30-emmc)
+ 		caldata_extract_mmc "factory" 0x0 0x1000
+ 		;;
 	esac
 	;;
 "mediatek/mt7986_eeprom_mt7975_dual.bin")
@@ -35,6 +40,10 @@ case "$FIRMWARE" in
 		ln -sf /tmp/tp_data/MT7986_EEPROM.bin \
 			/lib/firmware/$FIRMWARE
 		;;
+	glinet,gl-mt6000|\
+	jdcloud,re-cp-03)
+		caldata_extract_mmc "factory" 0x0 0x1000
+		;;
 	esac
 	;;
 "mediatek/mt7996/mt7992_eeprom.bin")
