--- a/target/linux/mediatek/image/filogic.mk
+++ b/target/linux/mediatek/image/filogic.mk
@@ -1206,22 +1206,22 @@ endif
 endef
 TARGET_DEVICES += xiaomi_mi-router-ax3000t-ubootmod
 
-define Device/xiaomi_mi-router-wr30u-stock
+define Device/xiaomi_mi-router-wr30u
   DEVICE_VENDOR := Xiaomi
-  DEVICE_MODEL := Mi Router WR30U (stock layout)
-  DEVICE_DTS := mt7981b-xiaomi-mi-router-wr30u-stock
+  DEVICE_MODEL := Mi Router WR30U
+  DEVICE_DTS := mt7981b-xiaomi-mi-router-wr30u
   DEVICE_DTS_DIR := ../dts
   UBINIZE_OPTS := -E 5
   BLOCKSIZE := 128k
   PAGESIZE := 2048
+  IMAGE_SIZE := 114688k
+  KERNEL_IN_UBI := 1
+  IMAGES += factory.bin
+  IMAGE/factory.bin := append-ubi | check-size $$$$(IMAGE_SIZE)
   DEVICE_PACKAGES := kmod-mt7915e kmod-mt7981-firmware mt7981-wo-firmware
-ifneq ($(CONFIG_TARGET_ROOTFS_INITRAMFS),)
-  ARTIFACTS := initramfs-factory.ubi
-  ARTIFACT/initramfs-factory.ubi := append-image-stage initramfs-kernel.bin | ubinize-kernel
-endif
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
 endef
-TARGET_DEVICES += xiaomi_mi-router-wr30u-stock
+TARGET_DEVICES += xiaomi_mi-router-wr30u
 
 define Device/xiaomi_mi-router-wr30u-ubootmod
   DEVICE_VENDOR := Xiaomi