--- a/target/linux/mediatek/dts/mt7981b-cetron-ct3003.dts
+++ b/target/linux/mediatek/dts/mt7981b-cetron-ct3003.dts
@@ -159,18 +159,9 @@
 
 			partition@580000 {
 				label = "ubi";
-				reg = <0x0580000 0x2000000>;
+				reg = <0x0580000 0x7000000>;
 			};
 
-			partition@2580000 {
-				label = "ubi_backup";
-				reg = <0x2580000 0x2000000>;
-			};
-
-			partition@4580000 {
-				label = "Config_backup";
-				reg = <0x4580000 0x0400000>;
-			};
 		};
 	};
 };
