--- a/target/linux/mediatek/dts/mt7981b-xiaomi-mi-router-ax3000t.dts
+++ b/target/linux/mediatek/dts/mt7981b-xiaomi-mi-router-ax3000t.dts
@@ -17,18 +17,12 @@
 &partitions {
 	// ubi_kernel is the ubi partition in stock.
 	partition@600000 {
-		label = "ubi_kernel";
-		reg = <0x600000 0x2200000>;
+		label = "ubi";
+		reg = <0x600000 0x7000000>;
 	};
 
-	/* ubi is the result of squashing
-	 * consecutive stock partitions:
-	 * - ubi1
-	 * - overlay
-	 * - data
-	 */
-	partition@2800000 {
-		label = "ubi";
-		reg = <0x2800000 0x4e00000>;
+	partition@7600000 {
+		label = "KF";
+		reg = <0x7600000 0x40000>;
 	};
 };