--- a/target/linux/mediatek/dts/mt7986a-ruijie-rg-x60-pro.dts
+++ b/target/linux/mediatek/dts/mt7986a-ruijie-rg-x60-pro.dts
@@ -199,7 +199,6 @@
 			partition@380000 {
 				label = "FIP";
 				reg = <0x380000 0x200000>;
-				read-only;
 			};
 
 			partition@580000 {
@@ -216,7 +215,7 @@
 
 			partition@680000 {
 				label = "ubi";
-				reg = <0x680000 0x3f00000>;
+				reg = <0x680000 0x6b00000>;
 			};
 		};
 	};