--- a/target/linux/mediatek/image/mt7622.mk
+++ b/target/linux/mediatek/image/mt7622.mk
@@ -438,18 +438,10 @@ define Device/xiaomi_redmi-router-ax6s
   UBINIZE_OPTS := -E 5
   BLOCKSIZE := 128k
   PAGESIZE := 2048
-  KERNEL := kernel-bin | gzip
-  KERNEL_INITRAMFS := kernel-bin | lzma | fit lzma $$(KDIR)/image-$$(firstword $$(DEVICE_DTS)).dtb with-initrd | pad-to 64k
+  KERNEL_SIZE := 6144k
   KERNEL_INITRAMFS_SUFFIX := -recovery.itb
-  IMAGES := sysupgrade.itb
-  IMAGE/sysupgrade.itb := append-kernel | fit gzip $$(KDIR)/image-$$(firstword $$(DEVICE_DTS)).dtb external-static-with-rootfs | append-metadata
-  ARTIFACTS := ubi-loader.itb
-  ARTIFACT/ubi-loader.itb := uboot-bin xiaomi_redmi-router-ax6s-ubi-loader | lzma | uboot-fit lzma
-ifneq ($(CONFIG_TARGET_ROOTFS_SQUASHFS),)
-  ARTIFACTS += factory.bin
-  ARTIFACT/factory.bin := uboot-bin xiaomi_redmi-router-ax6s-ubi-loader | lzma | uboot-fit lzma | pad-to 512k | ubinize-image fit squashfs-sysupgrade.itb
-endif
-  DEVICE_COMPAT_VERSION := 2.0
-  DEVICE_COMPAT_MESSAGE := Flash layout changes require a manual reinstall using factory.bin.
+  IMAGES += factory.bin
+  IMAGE/factory.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
 endef
 TARGET_DEVICES += xiaomi_redmi-router-ax6s

--- a/target/linux/mediatek/dts/mt7622-xiaomi-redmi-router-ax6s.dts
+++ b/target/linux/mediatek/dts/mt7622-xiaomi-redmi-router-ax6s.dts
@@ -22,8 +22,7 @@
 
 	chosen {
 		stdout-path = "serial0:115200n8";
-		rootdisk = <&ubi_rootfs>;
-		bootargs = "earlycon=uart8250,mmio32,0x11002000 swiotlb=512 ubi.block=0,fit root=/dev/fit0";
+		bootargs = "earlycon=uart8250,mmio32,0x11002000 swiotlb=512";
 	};
 
 	memory {
@@ -297,8 +296,8 @@
 			};
 
 			partition@2c0000 {
-				label = "ubi-loader";
-				reg = <0x2c0000 0x80000>;
+				label = "kernel";
+				reg = <0x2c0000 0x600000>;
 			};
 
 			/* ubi partition is the result of squashing
@@ -308,16 +307,9 @@
 			 * - overlay
 			 * - obr
 			 */
-			partition@340000 {
+			partition@8c0000 {
 				label = "ubi";
-				reg = <0x340000 0x7280000>;
-				compatible = "linux,ubi";
-
-				volumes {
-					ubi_rootfs: ubi-volume-fit {
-						volname = "fit";
-					};
-				};
+				reg = <0x8c0000 0x6d00000>;
 			};
 		};
 	};
