--- a/target/linux/generic/files/drivers/mtd/nand/mtk_bmt_nmbm.c
+++ b/target/linux/generic/files/drivers/mtd/nand/mtk_bmt_nmbm.c
@@ -2200,11 +2200,15 @@ static int nmbm_attach(struct nmbm_instance *ni)
 	if (ni->signature.nand_size != bmtd.total_blks << bmtd.blk_shift ||
 	    ni->signature.block_size != bmtd.blk_size ||
 	    ni->signature.page_size != bmtd.pg_size ||
-	    ni->signature.spare_size != bmtd.mtd->oobsize) {
+	    ni->signature.spare_size > bmtd.mtd->oobsize) {
 		nlog_err(ni, "NMBM configuration mismatch\n");
 		return -EINVAL;
 	}
 
+	if (ni->signature.spare_size < bmtd.mtd->oobsize)
+		nlog_warn(ni, "NMBM spare size (%u) is less than mtd oob size (%u)\n",
+			 ni->signature.spare_size, bmtd.mtd->oobsize);
+
 	success = nmbm_load_existing(ni);
 	if (!success)
 		return -ENODEV;
