--- a/target/linux/at91/image/sama5.mk
+++ b/target/linux/at91/image/sama5.mk
@@ -111,8 +111,8 @@ TARGET_DEVICES += microchip_sama5d2-ptc-ek
 
 define Device/microchip_sama5d3-xplained
   $(Device/evaluation-dtb)
-  DEVICE_VENDOR := Microchip
-  DEVICE_MODEL := SAMA5D3 Xplained
+  DEVICE_VENDOR := 玩客云
+  DEVICE_MODEL := OneCloud
   DEVICE_DTS := at91-sama5d3_xplained
   SUPPORTED_DEVICES := atmel,sama5d3-xplained
   KERNEL_SIZE := 6144k

--- a/target/linux/at91/image/Makefile
+++ b/target/linux/at91/image/Makefile
@@ -76,4 +76,15 @@ define Device/production-dtb
 	append-kernel | pad-to $$$$(KERNEL_SIZE) | append-ubi
 endef
 
+define Image/Build
+	if [ $(PROFILE_SANITIZED) == "microchip_sama5d3-xplained" ]; then \
+		export IMG_PREFIX="$(IMG_PREFIX)$(if $(PROFILE_SANITIZED),-$(PROFILE_SANITIZED))"; \
+		export BIN_DIR=$(BIN_DIR); \
+		export TARGET_ROOTFS_DIR=$(TARGET_DIR); \
+		cd $(TOPDIR)/../packit; \
+		./build.sh; \
+		gzip -f9n $(BIN_DIR)/$(IMG_PREFIX)$(if $(PROFILE_SANITIZED),-$(PROFILE_SANITIZED)).img || true; \
+	fi
+endef
+
 $(eval $(call BuildImage))