--- a/package/feeds/custom/netdata/Makefile
+++ b/package/feeds/custom/netdata/Makefile 
@@ -30,7 +30,7 @@ include $(INCLUDE_DIR)/package.mk
 define Package/netdata
   SECTION:=admin
   CATEGORY:=Administration
-  DEPENDS:=+zlib +libuuid +libuv +libmnl +libjson-c
+  DEPENDS:=+zlib +libuuid +libuv +libmnl +libjson-c +libopenssl
   TITLE:=Real-time performance monitoring tool
   URL:=https://www.netdata.cloud/
 endef
@@ -55,3 +55,2 @@ CONFIGURE_ARGS += \
 	--without-libcap \
-	--disable-https \
 	--disable-dbengine \
