--- feeds/custom/luci/luci-app-adguardhome/root/etc/init.d/AdGuardHome
+++ feeds/custom/luci/luci-app-adguardhome/root/etc/init.d/AdGuardHome
@@ -19,9 +19,7 @@ set_forward_dnsmasq()
 	addr="127.0.0.1#$PORT"
 	OLD_SERVER="`uci get dhcp.@dnsmasq[0].server 2>/dev/null`"
 	echo $OLD_SERVER | grep "^$addr" >/dev/null 2>&1
-	if [ $? -eq 0 ]; then
-		return
-	fi
+
 	uci delete dhcp.@dnsmasq[0].server 2>/dev/null
 	uci add_list dhcp.@dnsmasq[0].server=$addr
 	for server in $OLD_SERVER; do
@@ -42,9 +40,6 @@ stop_forward_dnsmasq()
 	addr="127.0.0.1#$OLD_PORT"
 	OLD_SERVER="`uci get dhcp.@dnsmasq[0].server 2>/dev/null`"
 	echo $OLD_SERVER | grep "^$addr" >/dev/null 2>&1
-	if [ $? -ne 0 ]; then
-		return
-	fi
 
 	uci del_list dhcp.@dnsmasq[0].server=$addr 2>/dev/null
 	addrlist="`uci get dhcp.@dnsmasq[0].server 2>/dev/null`"
@@ -490,6 +485,12 @@ start_service() {
 	echo "AdGuardHome service enabled"
 	echo "luci enable switch=$enabled"
 	(sleep 10 && [ -z "$(pgrep $binpath)" ] && logger "AdGuardHome" "no process in 10s cancel redirect" && _do_redirect 0 )&
+	[ "$(uci get shadowsocksr.@global[0].pdnsd_enable)" == 0 ] || {
+		uci set shadowsocksr.@global[0].pdnsd_enable='0'
+		uci del shadowsocksr.@global[0].tunnel_forward
+		uci commit shadowsocksr
+		service shadowsocksr reload
+	}
 }
 reload_service()
 {
