--- package/feeds/custom/luci-app-ssr-plus/Makefile
+++ package/feeds/custom/luci-app-ssr-plus/Makefile
@@ -52,6 +52,10 @@ LUCI_DEPENDS:=+shadowsocksr-libev-alt +ipset +ip-full +iptables-mod-tproxy +dnsm
 	+PACKAGE_$(PKG_NAME)_INCLUDE_Kcptun:kcptun-client \
 	+PACKAGE_$(PKG_NAME)_INCLUDE_ShadowsocksR_Server:shadowsocksr-libev-server
 
+define Package/luci-app-ssr-plus/conffiles
+/etc/config/shadowsocksr
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-unblockmusic/Makefile
+++ package/feeds/custom/luci-app-unblockmusic/Makefile
@@ -28,6 +28,10 @@ LUCI_PKGARCH:=all
 
 PKG_MAINTAINER:=lean
 
+define Package/luci-app-unblockmusic/conffiles
+/etc/config/unblockmusic
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-qbittorrent/Makefile
+++ package/feeds/custom/luci-app-qbittorrent/Makefile
@@ -14,4 +14,8 @@ PKG_NAME:=luci-app-qbittorrent
 
+define Package/luci-app-qbittorrent/conffiles
+/etc/config/qbittorrent
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-kodexplorer/Makefile
+++ package/feeds/custom/luci-app-kodexplorer/Makefile
@@ -13,4 +13,8 @@ LUCI_PKGARCH:=all
 
+define Package/luci-app-kodexplorer/conffiles
+/etc/config/kodexplorer
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-zerotier/Makefile
+++ package/feeds/custom/luci-app-zerotier/Makefile
@@ -14,4 +14,8 @@ LUCI_PKGARCH:=all
 
+define Package/luci-app-zerotier/conffiles
+/etc/config/zerotier
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-turboacc/Makefile
+++ package/feeds/custom/luci-app-turboacc/Makefile
@@ -44,6 +44,10 @@ PKG_CONFIG_DEPENDS:= \
 	CONFIG_PACKAGE_$(PKG_NAME)_INCLUDE_bbr-cca \
 	CONFIG_PACKAGE_$(PKG_NAME)_INCLUDE_dnsforwarder
 
+define Package/luci-app-turboacc/conffiles
+/etc/config/turboacc
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-adbyby-plus/Makefile
+++ package/feeds/custom/luci-app-adbyby-plus/Makefile
@@ -14,4 +14,8 @@ PKG_NAME:=luci-app-adbyby-plus
 
+define Package/luci-app-adbyby-plus/conffiles
+/etc/config/adbyby
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-autoreboot/Makefile
+++ package/feeds/custom/luci-app-autoreboot/Makefile
@@ -13,4 +13,8 @@ LUCI_PKGARCH:=all
 
+define Package/luci-app-autoreboot/conffiles
+/etc/config/autoreboot
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-usb-printer/Makefile
+++ package/feeds/custom/luci-app-usb-printer/Makefile
@@ -13,4 +13,8 @@ LUCI_DEPENDS:=+p910nd +kmod-usb-printer
 
+define Package/luci-app-usb-printer/conffiles
+/etc/config/usb_printer
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-vlmcsd/Makefile
+++ package/feeds/custom/luci-app-vlmcsd/Makefile
@@ -13,4 +13,8 @@ LUCI_DEPENDS:=+vlmcsd
 
+define Package/luci-app-vlmcsd/conffiles
+/etc/config/vlmcsd
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-softethervpn/Makefile
+++ package/feeds/custom/luci-app-softethervpn/Makefile
@@ -13,4 +13,8 @@ LUCI_PKGARCH:=all
 
+define Package/luci-app-softethervpn/conffiles
+/etc/config/softethervpn
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-xlnetacc/Makefile
+++ package/feeds/custom/luci-app-xlnetacc/Makefile
@@ -11,6 +11,10 @@ LUCI_DEPENDS:=+jshn +wget +openssl-util
 LUCI_PKGARCH:=all
 LUCI_DESCRIPTION=LuCI Support for XLNetAcc.
 
+define Package/luci-app-xlnetacc/conffiles
+/etc/config/xlnetacc
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/luci-app-openclash/Makefile
+++ package/feeds/custom/luci-app-openclash/Makefile
@@ -22,6 +22,10 @@ define Package/$(PKG_NAME)/description
     A LuCI support for clash
 endef
 
+define Package/luci-app-openclash/conffiles
+/etc/config/openclash
+endef
+
 define Package/$(PKG_NAME)/postinst
 endef

--- package/feeds/custom/luci-app-accesscontrol/Makefile
+++ package/feeds/custom/luci-app-accesscontrol/Makefile
@@ -15,4 +15,8 @@ PKG_NAME:=luci-app-accesscontrol
 
+define Package/luci-app-accesscontrol/conffiles
+/etc/config/mia
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- package/feeds/custom/vsftpd-alt/Makefile
+++ package/feeds/custom/vsftpd-alt/Makefile
@@ -33,7 +33,7 @@ define Package/vsftpd-alt/Default
 endef
 
 define Package/vsftpd-alt/conffiles
-/etc/vsftpd.conf
+/etc/config/vsftpd.conf
 endef
 
 #EXTRA_LDFLAGS:= -lcrypt -lcrypto -lssl
