--- a/feeds/custom/luci/luci-app-ssr-plus/Makefile
+++ b/feeds/custom/luci/luci-app-ssr-plus/Makefile
@@ -52,6 +52,10 @@ LUCI_DEPENDS:=+shadowsocksr-libev-alt +ipset +ip-full +iptables-mod-tproxy +dnsm
 	+PACKAGE_$(PKG_NAME)_INCLUDE_Kcptun:kcptun-client \
 	+PACKAGE_$(PKG_NAME)_INCLUDE_ShadowsocksR_Server:shadowsocksr-libev-server
 
+define Package/luci-app-ssr-plus/conffiles
+/etc/config/shadowsocksr
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- a/feeds/custom/luci/luci-app-unblockmusic/Makefile
+++ b/feeds/custom/luci/luci-app-unblockmusic/Makefile
@@ -28,6 +28,10 @@ LUCI_PKGARCH:=all
 
 PKG_MAINTAINER:=lean
 
+define Package/luci-app-unblockmusic/conffiles
+/etc/config/unblockmusic
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature
 
--- a/feeds/custom/luci/luci-app-amule/Makefile
+++ b/feeds/custom/luci/luci-app-amule/Makefile
@@ -9,6 +9,10 @@ LUCI_TITLE:=LuCI support for Amule
 LUCI_DEPENDS:=+amule
 LUCI_PKGARCH:=all
 
+define Package/luci-app-amule/conffiles
+/etc/config/amule
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- a/feeds/custom/luci/luci-app-qbittorrent/Makefile
+++ b/feeds/custom/luci/luci-app-qbittorrent/Makefile
@@ -12,6 +12,10 @@ PKG_NAME:=luci-app-qbittorrent
 PKG_VERSION=1.0
 PKG_RELEASE:=2
 
+define Package/luci-app-qbittorrent/conffiles
+/etc/config/qbittorrent
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature

--- a/feeds/custom/luci/luci-app-kodexplorer/Makefile
+++ b/feeds/custom/luci/luci-app-kodexplorer/Makefile
@@ -11,6 +11,10 @@ LUCI_PKGARCH:=all
 PKG_VERSION:=13
 PKG_DATE:=20200612
 
+define Package/luci-app-kodexplorer/conffiles
+/etc/config/kodexplorer
+endef
+
 include $(TOPDIR)/feeds/luci/luci.mk
 
 # call BuildPackage - OpenWrt buildroot signature
