--- feeds/packages/net/transmission-web-control/Makefile
+++ feeds/packages/net/transmission-web-control/Makefile
@@ -19,7 +19,7 @@ define Package/transmission-web-control
   SECTION:=net
   CATEGORY:=Network
   SUBMENU:=BitTorrent
-  DEPENDS:=@(PACKAGE_transmission-daemon-openssl||PACKAGE_transmission-daemon-mbedtls)
+  DEPENDS:=+transmission-daemon-openssl
   CONFLICTS:=transmission-web
   TITLE:=Transmission Web Control
   URL:=https://github.com/ronggang/transmission-web-control

--- feeds/packages/net/transmission/Makefile
+++ feeds/packages/net/transmission/Makefile
@@ -85,7 +85,7 @@ endef
 define Package/transmission-web
   $(call Package/transmission/template)
   TITLE+= (webinterface)
-  DEPENDS:=@(PACKAGE_transmission-daemon-openssl||PACKAGE_transmission-daemon-mbedtls)
+  DEPENDS:=+transmission-daemon-openssl
   PKGARCH:=all
 endef
 

--- package/feeds/custom/dnsforwarder/Makefile
+++ package/feeds/custom/dnsforwarder/Makefile
@@ -20,7 +20,7 @@ define Package/$(PKG_NAME)
   CATEGORY:=Network
   TITLE:=A simple DNS forwarder
   URL:=https://github.com/holmium/dnsforwarder
-  DEPENDS:=+wget +dnsmasq-full +libpthread +coreutils-base64
+  DEPENDS:=+wget +dnsmasq-full +libpthread +coreutils +coreutils-base64
 endef
 
 define Package/$(PKG_NAME)/description
