--- /feeds/custom/luci/luci-app-sfe/root/etc/init.d/sfe
+++ /feeds/custom/luci/luci-app-sfe/root/etc/init.d/sfe
@@ -146,3 +146,7 @@
 }
+
+service_triggers() {
+	procd_add_reload_trigger sfe
+}
 
 start_service() {

--- /feeds/custom/luci/luci-app-flowoffload/root/etc/init.d/flowoffload
+++ /feeds/custom/luci/luci-app-flowoffload/root/etc/init.d/flowoffload
@@ -152,3 +152,7 @@
 }
+
+service_triggers() {
+	procd_add_reload_trigger flowoffload
+}
 
 start_service() {
