--- a/feeds/luci/modules/luci-base/po/zh_Hans/base.po
+++ b/feeds/luci/modules/luci-base/po/zh_Hans/base.po
@@ -5053,6 +5053,12 @@ msgstr "禁止解析 IPv6 DNS 记录"
 msgid "Filter IPv6(AAAA) DNS Query Name Resolve"
 msgstr "过滤掉 IPv6(AAAA) ，只返回 IPv4 DNS 域名记录"
 
+msgid "Minimum TTL to send to clients"
+msgstr "客户端缓存的最小 DNS TTL"
+
+msgid "Modify DNS entries minimum TTL (max is 86400, 0 is no modify)"
+msgstr "修改发送到客户端的域名记录的 TTL 时间 (最大 86400, 0 表示不修改)"
+
 #: modules/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js:203
 msgid "Suppress logging"
 msgstr "不记录日志"

--- /feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
+++ /feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
@@ -343,2 +343,9 @@
 
+		o = s.taboption("advanced", form.Value, "mini_ttl",
+			_('Minimum TTL to send to clients'),
+			_('Modify DNS entries minimum TTL (max is 86400, 0 is no modify)'))
+		o.optional = true
+		o.datatype = "range(0,86400)"
+		o.placeholder = 0
+
 		s.taboption('tftp', form.Flag, 'enable_tftp',

--- /package/network/services/dnsmasq/files/dnsmasq.init
+++ /package/network/services/dnsmasq/files/dnsmasq.init
@@ -898,2 +898,3 @@
 	append_parm "$cfg" "local_ttl" "--local-ttl"
+	append_parm "$cfg" mini_ttl "--min-ttl"
 	append_parm "$cfg" "max_ttl" "--max-ttl"
