--- feeds/luci/applications/luci-app-opkg/luasrc/controller/opkg.lua
+++ feeds/luci/applications/luci-app-opkg/luasrc/controller/opkg.lua
@@ -41,3 +41,3 @@
 	local sys = require "luci.sys"
-	local cmd = { "/bin/opkg", "--force-removal-of-dependent-packages" }
+	local cmd = { "/bin/opkg", "--force-removal-of-dependent-packages", "--force-overwrite", "--force-checksum" }
 	local pkg = luci.http.formvalue("package")
@@ -60,2 +60,3 @@
 	luci.http.write_json(sys.process.exec(cmd, true, true))
+	luci.sys.exec("source /etc/profile.d/opkginstall.sh;opkg "..command.." >/dev/null 2>&1")

--- feeds/luci/modules/luci-mod-system/htdocs/luci-static/resources/view/system/flash.js
+++ feeds/luci/modules/luci-mod-system/htdocs/luci-static/resources/view/system/flash.js
@@ -203,6 +203,7 @@ return view.extend({
 			}, this, ev.target))
 			.then(L.bind(function(btn, res) {
 				var keep = E('input', { type: 'checkbox' }),
+				    keep_opkg = E('input', { type: 'checkbox' }),
 				    force = E('input', { type: 'checkbox' }),
 				    is_valid = res[1].valid,
 				    is_forceable = res[1].forceable,
@@ -220,6 +221,10 @@ return view.extend({
 				body.push(E('p', {}, E('label', { 'class': 'btn' }, [
 					keep, ' ', _('Keep settings and retain the current configuration')
 				])));
+				body.push(E('p', {}, E('label', { 'class': 'btn' }, [
+					keep_opkg, ' ', _('Retain the current packages')
+				])));
+					keep_opkg.checked = true;
 
 				if (!is_valid || is_too_big)
 					body.push(E('hr'));
@@ -257,7 +262,7 @@ return view.extend({
 
 				var cntbtn = E('button', {
 					'class': 'btn cbi-button-action important',
-					'click': ui.createHandlerFn(this, 'handleSysupgradeConfirm', btn, keep, force),
+					'click': ui.createHandlerFn(this, 'handleSysupgradeConfirm', btn, keep, keep_opkg, force),
 					'disabled': (!is_valid || is_too_big) ? true : null
 				}, [ _('Continue') ]);
 
@@ -282,7 +287,7 @@ return view.extend({
 			}, this, ev.target));
 	},
 
-	handleSysupgradeConfirm: function(btn, keep, force, ev) {
+	handleSysupgradeConfirm: function(btn, keep, keep_opkg, force, ev) {
 		btn.firstChild.data = _('Flashing…');
 
 		ui.showModal(_('Flashing…'), [
@@ -294,6 +299,9 @@ return view.extend({
 		if (!keep.checked)
 			opts.push('-n');
 
+		if (keep_opkg.checked)
+			opts.push('-k');
+
 		if (force.checked)
 			opts.push('--force');
 
