--- package/feeds/custom/libtorrent-rasterbar/Makefile
+++ package/feeds/custom/libtorrent-rasterbar/Makefile
@@ -68,7 +68,7 @@ endef
 
 define Package/libtorrent-rasterbar/install
 	$(INSTALL_DIR) $(1)/usr/lib
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libtorrent-rasterbar.so.* $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libtorrent-rasterbar.so* $(1)/usr/lib/
 endef
 
 $(eval $(call BuildPackage,libtorrent-rasterbar))
