--- a/target/linux/qualcommax/image/ipq50xx.mk
+++ b/target/linux/qualcommax/image/ipq50xx.mk
@@ -1,3 +1,28 @@
+define Device/glinet_gl-b3000
+  $(call Device/FitImage)
+  $(call Device/UbiFit)
+  SOC := ipq5018
+  DEVICE_VENDOR := GL.iNET
+  DEVICE_MODEL := GL-B3000
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  DEVICE_DTS_CONFIG := config@mp03.5-c1
+  DEVICE_PACKAGES := ath11k-firmware-qcn6122 ipq-wifi-gl-b3000
+endef
+TARGET_DEVICES += glinet_gl-b3000
+
+define Device/jdcloud_re-cs-03
+	$(call Device/FitImage)
+	$(call Device/EmmcImage)
+	SOC := ipq5018
+	BLOCKSIZE := 64k
+	KERNEL_SIZE := 6144k
+	DEVICE_VENDOR := JDCloud
+	DEVICE_MODEL := AX3000
+	DEVICE_DTS_CONFIG := config@mp03.5-c2
+endef
+TARGET_DEVICES += jdcloud_re-cs-03
+
 define Device/linksys_mx_atlas6
 	$(call Device/FitImageLzma)
 	DEVICE_VENDOR := Linksys

--- a/target/linux/qualcommax/ipq50xx/base-files/etc/board.d/02_network
+++ b/target/linux/qualcommax/ipq50xx/base-files/etc/board.d/02_network
@@ -7,6 +7,10 @@ ipq50xx_setup_interfaces()
 {
 	local board="$1"
 	case $board in
+		glinet,gl-b3000)
+		ucidef_set_interfaces_lan_wan "lan1 lan2" "wan"
+		;;
+		jdcloud,re-cs-03|\
 		linksys,mx2000|\
 		linksys,mx5500)
 			ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"

--- a/target/linux/qualcommax/ipq50xx/base-files/lib/upgrade/platform.sh
+++ b/target/linux/qualcommax/ipq50xx/base-files/lib/upgrade/platform.sh
@@ -10,6 +10,14 @@ platform_check_image() {
 
 platform_do_upgrade() {
 	case "$(board_name)" in
+ 	glinet,gl-b3000)
+		nand_do_upgrade "$1"
+		;;
+	jdcloud,re-cs-03)
+		CI_KERNPART="0:HLOS"
+		CI_ROOTPART="rootfs"
+		emmc_do_upgrade "$1"
+		;;
 	linksys,mx2000|\
 	linksys,mx5500)
 		platform_do_upgrade_linksys "$1"
@@ -19,3 +27,12 @@ platform_do_upgrade() {
 		;;
 	esac
 }
+
+platform_copy_config() {
+	case "$(board_name)" in
+	jdcloud,re-cs-03)
+		emmc_copy_config
+		;;
+	esac
+	return 0;
+}

--- a/target/linux/qualcommax/ipq50xx/base-files/etc/hotplug.d/firmware/11-ath11k-caldata
+++ b/target/linux/qualcommax/ipq50xx/base-files/etc/hotplug.d/firmware/11-ath11k-caldata
@@ -9,6 +9,15 @@ board=$(board_name)
 case "$FIRMWARE" in
 "ath11k/IPQ5018/hw1.0/cal-ahb-c000000.wifi.bin")
 	case "$board" in
+	glinet,gl-b3000)
+		caldata_extract "0:ART" 0x1000 0x20000
+		addr=$(mtd_get_mac_binary "0:ART" 0x6)
+		ath11k_patch_mac $(macaddr_add $addr 3) 0
+		ath11k_set_macflag
+		;;
+	jdcloud,re-cs-03)
+		caldata_extract_mmc "0:ART" 0x1000 0x20000
+		;;
 	linksys,mx2000|\
 	linksys,mx5500)
 		caldata_extract "0:ART" 0x1000 0x20000
@@ -21,6 +30,15 @@ case "$FIRMWARE" in
 	;;
 "ath11k/QCN6122/hw1.0/cal-ahb-b00a040.wifi1.bin")
 	case "$board" in
+	glinet,gl-b3000)
+		caldata_extract "0:ART" 0x26800 0x20000
+		addr=$(mtd_get_mac_binary "0:ART" 0x6)
+		ath11k_patch_mac $(macaddr_add $addr 4) 0
+		ath11k_set_macflag
+		;;
+	jdcloud,re-cs-03)
+		caldata_extract_mmc "0:ART" 0x26800 0x20000
+		;;
 	linksys,mx2000)
 		caldata_extract "0:ART" 0x26800 0x20000
 		label_mac=$(mtd_get_mac_ascii devinfo hw_mac_addr)
