--- a/target/linux/mediatek_gl/Makefile
+++ a/target/linux/mediatek_gl/Makefile
@@ -10,9 +10,9 @@
 
 KERNEL_PATCHVER:=5.4
 KERNEL_TESTING_PATCHVER:=5.4
 
 include $(INCLUDE_DIR)/target.mk
 DEFAULT_PACKAGES += \
-	kmod-leds-gpio kmod-gpio-button-hotplug
+	kmod-leds-gpio kmod-gpio-button-hotplug wpad-basic-wolfssl ethtool iwinfo wireless-tools kmod-mediatek_hnat mii_mgr mtkhnat_util regs fwdd kmod-mtfwd kmod-mtqos
 
 $(eval $(call BuildTarget))

--- a/target/linux/mediatek_gl/image/mt7981.mk
+++ b/target/linux/mediatek_gl/image/mt7981.mk
@@ -175,13 +175,13 @@
 define Device/glinet_gl-mt3000
   DEVICE_VENDOR := GL.iNet
   DEVICE_MODEL := GL-MT3000
   DEVICE_DTS := mt7981-gl-mt3000
   DEVICE_DTS_DIR := $(DTS_DIR)/mediatek
   SUPPORTED_DEVICES := glinet,mt3000-snand
-  DEVICE_PACKAGES := kmod-hwmon-pwmfan
+  DEVICE_PACKAGES := kmod-hwmon-pwmfan gl-sdk4-mtk-wifi-v2 kmod-conninfra-mt7981-ko kmod-mt-wifi-mt7981-ko kmod-warp-mt7981-ko kmod-gl-sdk4-fan 8021xd
   UBINIZE_OPTS := -E 5
   BLOCKSIZE := 128k
   PAGESIZE := 2048
   IMAGE_SIZE := 65536k
   KERNEL_IN_UBI := 1
   IMAGES := factory.img sysupgrade.tar
@@ -197,5 +197,5 @@
   DEVICE_DTS_DIR := $(DTS_DIR)/mediatek
   SUPPORTED_DEVICES := glinet,x3000-emmc
-  DEVICE_PACKAGES := kmod-hwmon-pwmfan mkf2fs kmod-mmc kmod-fs-f2fs gdisk
+  DEVICE_PACKAGES := kmod-hwmon-pwmfan mkf2fs kmod-mmc kmod-fs-f2fs gdisk mmc-utils losetup blockdev gl-sdk4-mtk-wifi-v2 kmod-conninfra-mt7981-ko kmod-mt-wifi-mt7981-ko kmod-warp-mt7981-ko kmod-gl-sdk4-fan kmod-shortcut-fe kmod-shortcut-fe-cm 8021xd
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-gl-metadata
 endef
@@ -208,5 +208,5 @@
   DEVICE_DTS_DIR := $(DTS_DIR)/mediatek
   SUPPORTED_DEVICES := glinet,xe3000-emmc
-  DEVICE_PACKAGES := kmod-hwmon-pwmfan mkf2fs kmod-mmc kmod-fs-f2fs gdisk
+  DEVICE_PACKAGES := kmod-hwmon-pwmfan mkf2fs kmod-mmc kmod-fs-f2fs gdisk mmc-utils losetup blockdev gl-sdk4-mtk-wifi-v2 kmod-conninfra-mt7981-ko kmod-mt-wifi-mt7981-ko kmod-warp-mt7981-ko kmod-gl-sdk4-fan kmod-shortcut-fe kmod-shortcut-fe-cm 8021xd
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-gl-metadata
 endef
@@ -215,10 +215,10 @@
 define Device/glinet_gl-mt2500
   DEVICE_VENDOR := GL.iNet
   DEVICE_MODEL := GL-MT2500
   DEVICE_DTS := mt7981-gl-mt2500
   SUPPORTED_DEVICES := glinet,mt2500-emmc
   DEVICE_DTS_DIR := $(DTS_DIR)/mediatek
-  DEVICE_PACKAGES := mkf2fs kmod-mmc kmod-fs-f2fs gdisk
+  DEVICE_PACKAGES := mkf2fs kmod-mmc kmod-fs-f2fs gdisk losetup e2fsprogs blockdev blkid kmod-nls-cp437 mtk_factory_rw mmc-utils
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-gl-metadata
 endef
 TARGET_DEVICES += glinet_gl-mt2500
