--- a/target/linux/qualcommax/image/ipq60xx.mk
+++ b/target/linux/qualcommax/image/ipq60xx.mk
@@ -10,6 +10,8 @@ define Device/cmiot_ax18
 	$(call Device/UbiFit)
 	DEVICE_VENDOR := CMIOT
 	DEVICE_MODEL := AX18
+	DEVICE_ALT0_VENDOR := ZN
+	DEVICE_ALT0_MODEL := M2
 	BLOCKSIZE := 128k
 	PAGESIZE := 2048
 	DEVICE_DTS_CONFIG := config@cp03-c1
@@ -127,7 +129,9 @@ define Device/xiaomi_rm1800
 	$(call Device/FitImage)
 	$(call Device/UbiFit)
 	DEVICE_VENDOR := Xiaomi
-	DEVICE_MODEL := RM1800 (AX5)
+	DEVICE_MODEL := AX1800
+	DEVICE_ALT0_VENDOR := Redmi
+	DEVICE_ALT0_MODEL := AX5
 	BLOCKSIZE := 128k
 	PAGESIZE := 2048
 	SOC := ipq6000
diff --git a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6000-ax18.dts b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6000-ax18.dts
index e887ff87a..b539d8dbb 100644
--- a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6000-ax18.dts
+++ b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6000-ax18.dts
@@ -213,6 +213,6 @@
 };
 
 &wifi {
-	status = "disabled";
+	status = "okay";
 	qcom,ath11k-fw-memory-mode = <2>;
 };
