--- a/package/feeds/custom/luci-app-bypass/luasrc/model/cbi/bypass/client-config.lua
+++ b/package/feeds/custom/luci-app-bypass/luasrc/model/cbi/bypass/client-config.lua
@@ -132,10 +130,8 @@ o:value("ssr", translate("ShadowsocksR"))
 if nixio.fs.access("/usr/bin/ss-redir") then
 o:value("ss", translate("Shadowsocks New Version"))
 end
-if nixio.fs.access("/usr/bin/xray") or nixio.fs.access("/usr/bin/v2ray") then
 o:value("vmess", translate("Vmess"))
 o:value("vless", translate("Vless"))
-end
 if nixio.fs.access("/usr/sbin/trojan-plus") then
 o:value("trojan", translate("Trojan"))
 end
@@ -407,12 +407,10 @@ o:depends({type = "vless", xtls = false})
 o:depends("type", "trojan")
 
 -- XTLS
-if nixio.fs.access("/usr/bin/xray") then
 o = s:option(Flag, "xtls", translate("XTLS"))
 o.rmempty = true
 o.default = "0"
 o:depends({type = "vless", transport = "tcp", tls = false})
-end
 
 -- Flow
 o = s:option(Value, "vless_flow", translate("Flow"))

--- a/package/feeds/custom/luci-app-bypass/root/etc/init.d/bypass
+++ b/package/feeds/custom/luci-app-bypass/root/etc/init.d/bypass
@@ -242,6 +242,18 @@
 	fi
 	NF_SERVER=$(uci_get_by_type global nf_server)
 	[ "$NF_SERVER" = $GLOBAL_SERVER ] && NF_SERVER=
+	L=0;M=$(uci_get_by_name $GLOBAL_SERVER type);N=$(uci_get_by_name $UDP_RELAY_SERVER type);O=$(uci_get_by_name $NF_SERVER type)
+	if [ "$M" = vmess -o "$M" = vless -o "$N" = vmess -o "$N" = vless -o "$O" = vmess -o "$O" = vless ];then
+		if [ ! -x /tmp/bypass/bin/xray ];then
+		[ -z "$switch_server" ] || L=1
+		if [ $L = 0 ];then
+			service_start $BIN_DIR/downbin --down $L
+			exit 1
+		else
+			$BIN_DIR/downbin --down $L
+		fi
+		fi
+	fi
 	start_rules && return 0 || return 1
 }
 
@@ -963,6 +975,7 @@
 }
 
 stop() {
+	kill -9 $(ps -w | grep $BIN_DIR/downbin | grep -v grep | awk '{print$1}') 2>/dev/null
 	kill -9 $(ps -w | grep by-rules | grep -v grep | awk '{print$1}') 2>/dev/null
 	kill -9 $(ps -w | grep gfw.b64 | grep -v grep | awk '{print$1}') 2>/dev/null
 	kill -9 $(ps -w | grep $BIN_DIR/checknetwork | grep -v grep | awk '{print$1}') 2>/dev/null
@@ -980,7 +993,7 @@
 	kill -9 $(ps -w | grep smartdns | grep -v grep | awk '{print$1}') 2>/dev/null
 	kill -9 $(ps -w | grep chinadns-ng | grep -v grep | awk '{print$1}') 2>/dev/null
 	ps -w | grep -v "grep" | grep "$TMP_PATH" | awk '{print $1}' | xargs kill -9 >/dev/null 2>&1 &
-	rm -rf $DNS_DIR $TMP_PATH /tmp/dnsmasq.d/dnsmasq-by.conf $CON_T /var/lock/bypass.lock
+	rm -rf $DNS_DIR $TMP_PATH /tmp/dnsmasq.d/dnsmasq-by.conf $CON_T /var/lock/bypass.lock /tmp/bypass/bin /usr/bin/xray
 	[ $run_mode = gfw -o $gfw_mode = 1 ] || rm -f /tmp/bypass/gfw.list
 	[ $run_mode = router ] || rm -f /tmp/bypass/china_v6.txt
 	if [ -z "$GLOBAL_SERVER" ]; then

--- /dev/null
+++ b/package/feeds/custom/luci-app-bypass/root/usr/share/bypass/downbin
@@ -0,0 +1,13 @@
+#!/bin/sh
+	[ "$1" = --down ] || exit 1
+	# 防止重复启动
+	[ -f /var/lock/bypass.lock ] && exit 1
+	touch /var/lock/bypass.lock
+	mkdir -p /tmp/bypass/bin
+		while ! curl -m 9 -Lfso /tmp/bypass/bin/xray https://cdn.jsdelivr.net/gh/garypang13/k2pxray/xray;do
+			sleep 2
+		done
+		chmod +x /tmp/bypass/bin/xray
+		ln -s /tmp/bypass/bin/xray /usr/bin/xray 2>/dev/null
+	rm -f /var/lock/bypass.lock
+	[ $2 = 0 ] && /etc/init.d/bypass start &
