--- a/target/linux/ramips/dts/mt7621_phicomm_k2p.dts
+++ b/target/linux/ramips/dts/mt7621_phicomm_k2p.dts
@@ -3,5 +3,5 @@
 #include "mt7621_phicomm_k2p.dtsi"
 
 &firmware {
-	reg = <0xa0000 0xf60000>;
+	reg = <0x50000 0xfb0000>;
 };

--- a/target/linux/ramips/dts/mt7621_phicomm_k2p.dtsi
+++ b/target/linux/ramips/dts/mt7621_phicomm_k2p.dtsi
@@ -80,14 +80,8 @@
 				read-only;
 			};
 
-			partition@50000 {
-				label = "permanent_config";
-				reg = <0x50000 0x50000>;
-				read-only;
-			};
-
-			firmware: partition@a0000 {
+			firmware: partition@50000 {
 				compatible = "denx,uimage";
 				label = "firmware";
 			};

--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -814,5 +814,5 @@
 
 define Device/phicomm_k2p
-  IMAGE_SIZE := 15744k
+  IMAGE_SIZE := 16064k
   DEVICE_VENDOR := Phicomm
   DEVICE_MODEL := K2P
