--- a/target/linux/ramips/dts/mt7621_phicomm_k2p.dts
+++ b/target/linux/ramips/dts/mt7621_phicomm_k2p.dts
@@ -39,15 +39,9 @@
 			};
 
 			partition@50000 {
-				label = "permanent_config";
-				reg = <0x50000 0x50000>;
-				read-only;
-			};
-
-			partition@a0000 {
 				compatible = "denx,uimage";
 				label = "firmware";
-				reg = <0xa0000 0xf60000>;
+				reg = <0x50000 0xfb0000>;
 			};
 		};
 	};

--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -814,5 +814,5 @@
 
 define Device/phicomm_k2p
-  IMAGE_SIZE := 15744k
+  IMAGE_SIZE := 16064k
   DEVICE_VENDOR := Phicomm
   DEVICE_MODEL := K2P
