--- a/target/linux/mediatek/image/mt7986.mk
+++ b/target/linux/mediatek/image/mt7986.mk
@@ -356,10 +356,11 @@ TARGET_DEVICES += mediatek_mt7986-fpga-ubi
 
 define Device/xiaomi_redmi-router-ax6000
   DEVICE_VENDOR := Xiaomi
-  DEVICE_MODEL := Redmi Router AX6000
+  DEVICE_MODEL := Redmi Router AX6000 (Uboot专用)
   DEVICE_DTS := mt7986a-xiaomi-redmi-router-ax6000
   DEVICE_DTS_DIR := $(DTS_DIR)/mediatek
   DEVICE_PACKAGES := kmod-leds-ws2812b
+  DEVICE_PACKAGES := luci-app-mtk mii_mgr wifi-profile switch regs kmod-warp kmod-mt_wifi kmod-mediatek_hnat kmod-conninfra datconf-lua
   SUPPORTED_DEVICES := xiaomi,redmi-router-ax6000
   UBINIZE_OPTS := -E 5
   BLOCKSIZE := 128k
@@ -374,10 +375,11 @@ TARGET_DEVICES += xiaomi_redmi-router-ax6000
 
 define Device/xiaomi_redmi-router-ax6000-stock
   DEVICE_VENDOR := Xiaomi
-  DEVICE_MODEL := Redmi Router AX6000 (stock layout)
+  DEVICE_MODEL := Redmi Router AX6000 (原厂布局)
   DEVICE_DTS := mt7986a-xiaomi-redmi-router-ax6000-stock
   DEVICE_DTS_DIR := $(DTS_DIR)/mediatek
   DEVICE_PACKAGES := kmod-leds-ws2812b
+  DEVICE_PACKAGES := luci-app-mtk mii_mgr wifi-profile switch regs kmod-warp kmod-mt_wifi kmod-mediatek_hnat kmod-conninfra datconf-lua
   SUPPORTED_DEVICES := xiaomi,redmi-router-ax6000-stock
   UBINIZE_OPTS := -E 5
   BLOCKSIZE := 128k