--- a/include/image.mk
+++ b/include/image.mk
@@ -609,6 +609,7 @@ define Device/Build/image
 	IMAGE_TYPE=$(word 1,$(subst ., ,$(2))) \
 	IMAGE_FILESYSTEM="$(1)" \
 	IMAGE_PREFIX="$(IMAGE_PREFIX)" \
+	KERNEL_INITRAMFS_IMAGE="$(subst $(IMG_PREFIX_EXTRA),,$(KERNEL_INITRAMFS_IMAGE))" \
 	DEVICE_VENDOR="$(DEVICE_VENDOR)" \
 	DEVICE_MODEL="$(DEVICE_MODEL)" \
 	DEVICE_VARIANT="$(DEVICE_VARIANT)" \

--- a/scripts/json_add_image_info.py
+++ b/scripts/json_add_image_info.py
@@ -48,6 +48,7 @@ def get_titles():
     "profiles": {
         device_id: {
             "image_prefix": getenv("IMAGE_PREFIX"),
+            "image_initramfs": getenv("KERNEL_INITRAMFS_IMAGE"),
             "images": [
                 {
                     "type": getenv("IMAGE_TYPE"),

--- a/scripts/json_overview_image_info.py
+++ b/scripts/json_overview_image_info.py
@@ -47,7 +47,7 @@ def get_initial_output(image_info):
 
 
 if output:
-    default_packages, output["arch_packages"] = run(
+    default_packages, output["arch_packages"], output["kernel_version"] = run(
         [
             "make",
             "--no-print-directory",
@@ -55,6 +55,7 @@ def get_initial_output(image_info):
             "target/linux/",
             "val.DEFAULT_PACKAGES",
             "val.ARCH_PACKAGES",
+            "val.LINUX_VERSION",
         ],
         stdout=PIPE,
         stderr=PIPE,

--- a/package/mtk/applications/luci-app-mtk/luasrc/view/admin_mtk/mtk_wifi_vif_cfg.htm
+++ b/package/mtk/applications/luci-app-mtk/luasrc/view/admin_mtk/mtk_wifi_vif_cfg.htm
@@ -581,9 +581,7 @@ <h2><%:Interface Configurations%> - <%=vifname and devname.."@"..vifname or devn
         </table>
         <h3><%:Access control MAC address list%></h3>
         <textarea name="__maclist" id="__maclist" style="width:98%; height: 100px;"><%
-        local list = cfgs["AccessControlList"..(vifidx-1)] or ""
-        print(table.concat(list:split(";"), "\n"))
-        %></textarea>
+        local list = cfgs["AccessControlList"..(vifidx-1)] or "" %><%=table.concat(list:split(";"), "\n")%></textarea>
         </fieldset>
 
         <% if string.split(cfgs.WirelessMode,";")[1] == "16" or string.split(cfgs.WirelessMode,";")[1] == "17" or string.split(cfgs.WirelessMode,";")[1] == "18" then %>
