--- a/include/target.mk
+++ b/include/target.mk
@@ -46,28 +46,27 @@ DEFAULT_PACKAGES.router:=\
 	dnsmasq-full \
 	firewall \
 	iptables \
-	ipv6helper \
+	ip6tables \
+	kmod-ipt-nat \
+	kmod-ipt-nat6 \
+	kmod-ipt-offload \
+	odhcp6c \
+	odhcpd-ipv6only \
 	ppp \
 	ppp-mod-pppoe
-# For easy usage
-DEFAULT_PACKAGES.tweak:=\
-	block-mount \
-	default-settings-chn \
-	kmod-ipt-raw \
-	kmod-nf-nathelper \
-	kmod-nf-nathelper-extra \
-	luci \
-	luci-app-filetransfer \
-	luci-app-turboacc \
-	luci-compat \
-	luci-lib-base \
-	luci-lib-fs \
-	luci-lib-ipkg
 
 ifneq ($(DUMP),)
   all: dumpinfo
 endif
 
+ifeq ($(ARCH),arm)
+  DEFAULT_PACKAGES+=luci-app-cpufreq
+endif
+
+ifeq ($(ARCH),aarch64)
+  DEFAULT_PACKAGES+=luci-app-cpufreq
+endif
+
 target_conf=$(subst .,_,$(subst -,_,$(subst /,_,$(1))))
 ifeq ($(DUMP),)
   PLATFORM_DIR:=$(TOPDIR)/target/linux/$(BOARD)