--- a/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rt-ac42u.dts
+++ b/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rt-ac42u.dts
@@ -225,38 +225,13 @@
 			#size-cells = <1>;
 
 			partition@0 {
-				label = "SBL1";
-				reg = <0x00000000 0x00080000>;
-				read-only;
-			};
-			partition@80000 {
-				label = "MIBIB";
-				reg = <0x00080000 0x00080000>;
-				read-only;
-			};
-			partition@100000 {
-				label = "QSEE";
-				reg = <0x00100000 0x00100000>;
-				read-only;
-			};
-			partition@200000 {
-				label = "CDT";
-				reg = <0x00200000 0x00080000>;
-				read-only;
-			};
-			partition@280000 {
-				label = "APPSBL";
-				reg = <0x00280000 0x00140000>;
-				read-only;
-			};
-			partition@3C0000 {
-				label = "APPSBLENV";
-				reg = <0x003C0000 0x00040000>;
+				label = "Bootloader";
+				reg = <0x000000000000 0x000000400000>;
 				read-only;
 			};
 			partition@400000 {
-				label = "ubi";
-				reg = <0x00400000 0x07C00000>;
+				label = "UBI_DEV";
+				reg = <0x000000400000 0x000007C00000>;
 			};
 		};
 	};