--- a/target/linux/ipq40xx/image/generic.mk
+++ b/target/linux/ipq40xx/image/generic.mk
@@ -694,7 +694,7 @@ TARGET_DEVICES += meraki_mr33
 define Device/mobipromo_cm520-79f
 	$(call Device/FitzImage)
 	$(call Device/UbiFit)
-	DEVICE_VENDOR := MobiPromo
+	DEVICE_VENDOR := MobiPromo星际宝盒
 	DEVICE_MODEL := CM520-79F
 	SOC := qcom-ipq4019
 	BLOCKSIZE := 128k
@@ -834,7 +834,7 @@ TARGET_DEVICES += openmesh_a62
 define Device/p2w_r619ac
 	$(call Device/FitzImage)
 	$(call Device/UbiFit)
-	DEVICE_VENDOR := P&W
+	DEVICE_VENDOR := P&W竞斗云
 	DEVICE_MODEL := R619AC
 	SOC := qcom-ipq4019
 	DEVICE_DTS_CONFIG := config@10