--- a/package/base-files/files/etc/board.d/99-default_network
+++ b/package/base-files/files/etc/board.d/99-default_network
@@ -8,8 +8,8 @@ board_config_update
 
 json_is_a network object && exit 0
 
-ucidef_set_interface_lan 'eth0'
-[ -d /sys/class/net/eth1 ] && ucidef_set_interface_wan 'eth1'
+ucidef_set_interface_wan 'eth0'
+[ -d /sys/class/net/eth1 ] && ucidef_set_interface_lan 'eth1'
 
 board_config_flush
