--- a/target/linux/ramips/dts/mt7621_xiaomi_mi-router-4a-common.dtsi
+++ b/target/linux/ramips/dts/mt7621_xiaomi_mi-router-4a-common.dtsi
@@ -54,40 +54,16 @@
 				read-only;
 			};
 
-			partition@40000 {
-				label = "Bdata";
-				reg = <0x40000 0x10000>;
-				read-only;
-			};
-
-			factory: partition@50000 {
+			factory: partition@40000 {
 				label = "factory";
-				reg = <0x50000 0x10000>;
-				read-only;
-			};
-
-			partition@60000 {
-				label = "crash";
-				reg = <0x60000 0x10000>;
-				read-only;
-			};
-
-			partition@70000 {
-				label = "cfg_bak";
-				reg = <0x70000 0x10000>;
-				read-only;
-			};
-
-			partition@80000 {
-				label = "overlay";
-				reg = <0x80000 0x100000>;
+				reg = <0x40000 0x10000>;
 				read-only;
 			};
 
-			firmware: partition@180000 {
+			partition@50000 {
 				compatible = "denx,uimage";
 				label = "firmware";
-				reg = <0x180000 0xe80000>;
+				reg = <0x50000 0xfb0000>;
 			};
 		};
 	};

--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -1963,7 +1963,7 @@ TARGET_DEVICES += xiaomi_mi-router-3g
 define Device/xiaomi_mi-router-3g-v2
   $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
-  IMAGE_SIZE := 14848k
+  IMAGE_SIZE := 16064k
   DEVICE_VENDOR := Xiaomi
   DEVICE_MODEL := Mi Router 3G
   DEVICE_VARIANT := v2
@@ -2003,11 +2003,11 @@ TARGET_DEVICES += xiaomi_mi-router-4
 define Device/xiaomi_mi-router-4a-gigabit
   $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
-  IMAGE_SIZE := 14848k
+  IMAGE_SIZE := 16064k
   DEVICE_VENDOR := Xiaomi
   DEVICE_MODEL := Mi Router 4A
   DEVICE_VARIANT := Gigabit Edition
-  DEVICE_PACKAGES := kmod-mt7603 kmod-mt76x2
+  DEVICE_PACKAGES := kmod-mt7603 kmod-mt76x2 wpad-mini -wpad-basic-wolfssl -coremark -htop -bash -openssh-sftp-server
 endef
 TARGET_DEVICES += xiaomi_mi-router-4a-gigabit
 