--- a/target/linux/ramips/dts/mt7621_jcg_q20.dts
+++ b/target/linux/ramips/dts/mt7621_jcg_q20.dts
@@ -50,22 +50,6 @@
 			linux,code = <KEY_WPS_BUTTON>;
 		};
 	};
-
-	ubi-concat {
-		compatible = "mtd-concat";
-		devices = <&ubiconcat0 &ubiconcat1>;
-
-		partitions {
-			compatible = "fixed-partitions";
-			#address-cells = <1>;
-			#size-cells = <1>;
-
-			partition@0 {
-				label = "ubi";
-				reg = <0x0 0x5900000>;
-			};
-		};
-	};
 };
 
 &nand {
@@ -79,12 +63,18 @@
 		partition@0 {
 			label = "Bootloader";
 			reg = <0x0 0x80000>;
-			read-only;
 		};
 
 		partition@80000 {
-			label = "Config";
-			reg = <0x80000 0x80000>;
+			label = "Nvram";
+			reg = <0x80000 0x40000>;
+			read-only;
+		};
+
+		partition@c0000 {
+			label = "Bdata";
+			reg = <0xc0000 0x40000>;
+			read-only;
 		};
 
 		factory: partition@100000 {
@@ -94,40 +84,26 @@
 		};
 
 		partition@180000 {
-			label = "kernel";
-			reg = <0x180000 0x400000>;
-		};
-
-		ubiconcat0: partition@580000 {
-			label = "ubiconcat0";
-			reg = <0x580000 0x1c00000>;
-		};
-
-		partition@2180000 {
-			label = "firmware_backup";
-			reg = <0x2180000 0x2000000>;
-		};
-
-		partition@4180000 {
-			label = "rootfs_data_back";
-			reg = <0x4180000 0x80000>;
+			label = "crash";
+			reg = <0x180000 0x40000>;
 			read-only;
 		};
 
-		partition@4200000 {
-			label = "nvram_config";
-			reg = <0x4200000 0x80000>;
+		partition@1c0000 {
+			label = "crash_log";
+			reg = <0x1c0000 0x40000>;
 			read-only;
 		};
 
-		ubiconcat1: partition@4280000 {
-			label = "ubiconcat1";
-			reg = <0x4280000 0x3d00000>;
+		partition@200000 {
+			label = "kernel";
+			reg = <0x200000 0x400000>;
 		};
 
-		/*
-		 * last 512 KiB are for the bad block table
-		 */
+		partition@600000 {
+			label = "ubi";
+			reg = <0x600000 0x7980000>;
+		};
 	};
 };
 