diff --git a/target/linux/ramips/dts/mt7621_phicomm_k2p.dts b/target/linux/ramips/dts/mt7621_phicomm_k2p.dtsi
similarity index 94%
rename from target/linux/ramips/dts/mt7621_phicomm_k2p.dts
rename to target/linux/ramips/dts/mt7621_phicomm_k2p.dtsi
index b18d12608acc7..8f72ac2a48280 100644
--- a/target/linux/ramips/dts/mt7621_phicomm_k2p.dts
+++ b/target/linux/ramips/dts/mt7621_phicomm_k2p.dtsi
@@ -4,9 +4,6 @@
 #include <dt-bindings/input/input.h>
 
 / {
-	compatible = "phicomm,k2p", "mediatek,mt7621-soc";
-	model = "Phicomm K2P";
-
 	aliases {
 		led-boot = &led_blue;
 		led-failsafe = &led_blue;
@@ -47,7 +44,7 @@
 &spi0 {
 	status = "okay";
 
-	flash@0 {
+	flash: flash@0 {
 		compatible = "jedec,spi-nor";
 		reg = <0>;
 		spi-max-frequency = <50000000>;
@@ -81,10 +78,9 @@
 				read-only;
 			};
 
-			partition@a0000 {
+			firmware: partition@a0000 {
 				compatible = "denx,uimage";
 				label = "firmware";
-				reg = <0xa0000 0xf60000>;
 			};
 		};
 	};
diff --git a/target/linux/ramips/dts/mt7621_phicomm_k2p.dts b/target/linux/ramips/dts/mt7621_phicomm_k2p.dts
new file mode 100644
index 0000000000000..5bad6f2d9562d
--- /dev/null
+++ b/target/linux/ramips/dts/mt7621_phicomm_k2p.dts
@@ -0,0 +1,10 @@
+#include "mt7621_phicomm_k2p.dtsi"
+
+/ {
+	compatible = "phicomm,k2p", "mediatek,mt7621-soc";
+ 	model = "Phicomm K2P";
+};
+
+&firmware {
+	reg = <0xa0000 0x1f60000>;
+};
diff --git a/target/linux/ramips/dts/mt7621_phicomm_k2p-32m.dts b/target/linux/ramips/dts/mt7621_phicomm_k2p-32m.dts
new file mode 100644
index 0000000000000..5bad6f2d9562d
--- /dev/null
+++ b/target/linux/ramips/dts/mt7621_phicomm_k2p-32m.dts
@@ -0,0 +1,14 @@
+#include "mt7621_phicomm_k2p.dtsi"
+
+/ {
+	compatible = "phicomm,k2p-32m", "mediatek,mt7621-soc";
+	model = "Phicomm K2P (32M)";
+};
+
+&firmware {
+	reg = <0xa0000 0x1f60000>;
+};
+
+&flash {
+	broken-flash-reset;
+};
--- a/target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac
+++ b/target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac
@@ -28,7 +28,8 @@ case "$board" in
 				> /sys${DEVPATH}/macaddress
 		;;
 	dlink,dir-853-r1|\
-	phicomm,k2p)
+	phicomm,k2p|\
+	phicomm,k2p-32m)
 		if [ "$PHYNBR" = "0" ]; then
 			base_mac=$(macaddr_add "$(mtd_get_mac_binary factory 0x4)" -1)
 			macaddr_setbit_la "$base_mac" > /sys${DEVPATH}/macaddress
