diff --git a/target/linux/ramips/mt7621/base-files/etc/board.d/01_leds b/target/linux/ramips/mt7621/base-files/etc/board.d/01_leds
index aad2e32b36210..a6a4707fc295d 100644
--- a/target/linux/ramips/mt7621/base-files/etc/board.d/01_leds
+++ b/target/linux/ramips/mt7621/base-files/etc/board.d/01_leds
@@ -157,6 +157,16 @@ zbtlink,zbt-wg1608-16m)
 	ucidef_set_led_netdev "lan4" "LAN4" "green:lan-4" "lan4"
 	ucidef_set_led_netdev "wan" "WAN" "green:wan" "wan"
 	;;
+zte,e8820s)
+	ucidef_set_led_netdev "sys" "SYS_LED" "white:sys" "eth0" "tx rx"
+	ucidef_set_led_timer "power" "POWER_LED" "white:sys" "100" "8000"
+	ucidef_set_led_netdev "wlan2g" "WiFi 2.4GHz" "white:wlan2g" "ra0"
+	ucidef_set_led_netdev "wlan5g" "WiFi 5GHz" "white:wlan5g" "rai0"
+	;;
+zte,e8820v2)
+	ucidef_set_led_netdev "sys" "SYS_LED" "white:sys" "eth0" "tx rx"
+	ucidef_set_led_timer "power" "POWER_LED" "white:power" "100" "8000"
+	;;
 esac
 
 board_config_flush
