--- a/target/linux/sunxi/image/cortexa7.mk
+++ b/target/linux/sunxi/image/cortexa7.mk
@@ -47,7 +47,7 @@ TARGET_DEVICES += friendlyarm_nanopi-neo-air
 define Device/friendlyarm_nanopi-r1
   DEVICE_VENDOR := FriendlyARM
   DEVICE_MODEL := NanoPi R1
-  DEVICE_PACKAGES := kmod-usb-net-rtl8152 kmod-leds-gpio \
+  DEVICE_PACKAGES := kmod-usb-net-rtl8152 kmod-leds-gpio kmod-rtl8189es \
 	kmod-brcmfmac cypress-firmware-43430-sdio wpad-basic-wolfssl
   SOC := sun8i-h3
 endef