diff -rNEZbwBdu3 a/src/dnsmasq.h b/src/dnsmasq.h
--- a/src/dnsmasq.h	2020-04-09 00:32:53.000000000 +0800
+++ b/src/dnsmasq.h	2020-06-05 23:03:45.941176855 +0800
@@ -269,7 +269,8 @@
 #define OPT_IGNORE_CLID    59
 #define OPT_SINGLE_PORT    60
 #define OPT_LEASE_RENEW    61
-#define OPT_LAST           62
+#define OPT_FILTER_AAAA    62
+#define OPT_LAST           63
 
 #define OPTION_BITS (sizeof(unsigned int)*8)
 #define OPTION_SIZE ( (OPT_LAST/OPTION_BITS)+((OPT_LAST%OPTION_BITS)!=0) )
diff -rNEZbwBdu3 a/src/option.c b/src/option.c
--- a/src/option.c	2020-04-09 00:32:53.000000000 +0800
+++ b/src/option.c	2020-06-05 23:06:29.880350271 +0800
@@ -167,6 +167,7 @@
 #define LOPT_SINGLE_PORT   359
 #define LOPT_SCRIPT_TIME   360
 #define LOPT_PXE_VENDOR    361
+#define LOPT_FILTER_AAAA   362
  
 #ifdef HAVE_GETOPT_LONG
 static const struct option opts[] =  
@@ -339,6 +340,7 @@
     { "dumpfile", 1, 0, LOPT_DUMPFILE },
     { "dumpmask", 1, 0, LOPT_DUMPMASK },
     { "dhcp-ignore-clid", 0, 0,  LOPT_IGNORE_CLID },
+    { "filter-aaaa", 0, 0, LOPT_FILTER_AAAA },
     { NULL, 0, 0, 0 }
   };
 
@@ -518,6 +520,7 @@
   { LOPT_DUMPFILE, ARG_ONE, "<path>", gettext_noop("Path to debug packet dump file"), NULL },
   { LOPT_DUMPMASK, ARG_ONE, "<hex>", gettext_noop("Mask which packets to dump"), NULL },
   { LOPT_SCRIPT_TIME, OPT_LEASE_RENEW, NULL, gettext_noop("Call dhcp-script when lease expiry changes."), NULL },
+  { LOPT_FILTER_AAAA, OPT_FILTER_AAAA, NULL, gettext_noop("Filter all AAAA requests."), NULL },
   { 0, 0, NULL, NULL, NULL }
 }; 
 
diff -rNEZbwBdu3 a/src/rfc1035.c b/src/rfc1035.c
--- a/src/rfc1035.c	2020-04-09 00:32:53.000000000 +0800
+++ b/src/rfc1035.c	2020-06-05 23:08:46.476106541 +0800
@@ -1955,6 +1955,15 @@
 	    }
 	}
 
+      //patch to filter aaaa forwards
+      if (qtype == T_AAAA && option_bool(OPT_FILTER_AAAA) ){
+        //return a null reply
+        ans = 1;
+        if (!dryrun) log_query(F_CONFIG | F_IPV6 | F_NEG, name, &addr, NULL);
+        break;
+      }
+      //end of patch
+
       if (!ans)
 	return 0; /* failed to answer a question */
     }
