--- a/package/feeds/xywrt/base-files/Makefile
+++ b/package/feeds/xywrt/base-files/Makefile
@@ -205,6 +205,9 @@ define Package/base-files/install
 
 	$(if $(CONFIG_TARGET_PREINIT_DISABLE_FAILSAFE), \
 		rm -f $(1)/etc/banner.failsafe,)
+
+	$(if $(CONFIG_PACKAGE_wifi-profile), \
+		rm -f $(1)/sbin/wifi)
 endef
 
 ifneq ($(DUMP),1)

--- a/package/feeds/xywrt/base-files/files/etc/hotplug.d/net/00-sysctl
+++ b/package/feeds/xywrt/base-files/files/etc/hotplug.d/net/00-sysctl
@@ -6,4 +6,6 @@ if [ "$ACTION" = add ]; then
 		sed -ne "/^[[:space:]]*net\..*\.$DEVICENAME\./p" "$CONF" | \
 			sysctl -e -p - | logger -t sysctl
 	done
+
+	[ -f /sbin/smp.sh ] && /sbin/smp.sh
 fi
