--- a/target/linux/mediatek/image/mt7981.mk
+++ b/target/linux/mediatek/image/mt7981.mk
@@ -190,10 +190,11 @@ endef
 TARGET_DEVICES += mt7981-360-t7
 
 define Device/mt7981-360-t7-108M
-  DEVICE_VENDOR := MediaTek
+  DEVICE_VENDOR := Qihoo
   DEVICE_MODEL := 360 T7 (with 108M ubi)
   DEVICE_DTS := mt7981-360-t7-108M
   DEVICE_DTS_DIR := $(DTS_DIR)/mediatek
+  DEVICE_PACKAGES := luci-app-mtk mii_mgr wifi-profile switch regs kmod-warp kmod-mt_wifi kmod-mediatek_hnat kmod-conninfra datconf-lua
   SUPPORTED_DEVICES := 360,t7
   UBINIZE_OPTS := -E 5
   BLOCKSIZE := 128k