--- a/package/kernel/mac80211/broadcom.mk
+++ b/package/kernel/mac80211/broadcom.mk
@@ -432,6 +432,7 @@ define KernelPackage/brcmfmac/config
 
 	config BRCMFMAC_SDIO
 		bool "Enable SDIO bus interface support"
+		default y if TARGET_amlogic
 		default y if TARGET_bcm27xx
 		default y if TARGET_imx_cortexa7
 		default y if TARGET_starfive

--- a/target/linux/amlogic/base-files/etc/board.d/02_network
+++ b/target/linux/amlogic/base-files/etc/board.d/02_network
@@ -12,7 +12,7 @@ amlogic_setup_interfaces()
 
 	case "$board" in
 	*)
-		ucidef_set_interface_lan 'eth0' 'dhcp'
+		ucidef_set_interface_lan 'eth0'
 		;;
 	esac
 }
