--- a/package/feeds/packages/adblock/files/adblock.init
+++ b/package/feeds/packages/adblock/files/adblock.init
@@ -63,6 +63,7 @@ start_service()
 reload_service()
 {
 	rc_procd start_service reload
+	/etc/init.d/adblock report >/dev/null 2>&1
 }
 
 stop_service()
@@ -73,6 +74,7 @@ stop_service()
 restart()
 {
 	rc_procd start_service restart
+	/etc/init.d/adblock report >/dev/null 2>&1
 }
 
 susp
