--- a/package/feeds/custom/luci-app-adguardhome/root/etc/init.d/AdGuardHome
+++ b/package/feeds/custom/luci-app-adguardhome/root/etc/init.d/AdGuardHome
@@ -28,7 +28,7 @@ set_forward_dnsmasq()
 		if [ "$server" = "$addr" ]; then
 			continue
 		fi
-		uci add_list dhcp.@dnsmasq[0].server=$server
+		# uci add_list dhcp.@dnsmasq[0].server=$server
 	done
 	uci delete dhcp.@dnsmasq[0].resolvfile 2>/dev/null
 	uci set dhcp.@dnsmasq[0].noresolv=1
