--- a/package/base-files/files/etc/shinit
+++ b/package/base-files/files/etc/shinit
@@ -2,6 +2,8 @@
 [ -x /usr/bin/vim ] && alias vi=vim || alias vim=vi
 
 alias ll='ls -alF --color=auto'
+alias reboot='(/bin/busybox reboot &);sleep 3;/bin/busybox reboot -f'
+PS1='\[\033[35;1m\]\u\[\033[0m\]@\[\033[31;1m\]\h \[\e[36m\]\@\[\e[m\] \[\033[32;1m\]\W\[\033[0m\]\[\033[31m\]\$\[\033[0m\] '
 
 [ -z "$KSH_VERSION" -o \! -s /etc/mkshrc ] || . /etc/mkshrc
