--- a/package/feeds/luci/luci-mod-system/htdocs/luci-static/resources/view/system/crontab.js
+++ b/package/feeds/luci/luci-mod-system/htdocs/luci-static/resources/view/system/crontab.js
@@ -16,6 +16,7 @@ return view.extend({
 		return fs.write('/etc/crontabs/root', value).then(function(rc) {
 			document.querySelector('textarea').value = value;
 			ui.addNotification(null, E('p', _('Contents have been saved.')), 'info');
+			fs.exec('/etc/init.d/cron', [ 'restart' ]);
 		}).catch(function(e) {
 			ui.addNotification(null, E('p', _('Unable to save contents: %s').format(e.message)));
 		});

--- a/package/feeds/luci/luci-mod-system/root/usr/share/rpcd/acl.d/luci-mod-system.json
+++ b/package/feeds/luci/luci-mod-system/root/usr/share/rpcd/acl.d/luci-mod-system.json
@@ -72,6 +72,7 @@
 		},
 		"write": {
 			"file": {
+				"/etc/init.d/cron": [ "exec" ],
 				"/etc/crontabs/root": [ "write" ]
 			},
 			"ubus": {
