--- a/package/feeds/luci/luci-mod-network/htdocs/luci-static/resources/view/network/interfaces.js	2020-08-11 17:54:35.854075042 +0800
+++ b/package/feeds/luci/luci-mod-network/htdocs/luci-static/resources/view/network/interfaces.js	2020-08-17 20:32:46.229218269 +0800
@@ -867,6 +867,10 @@
 		o.datatype = 'cidr6';
 
 
+		o = s.option(form.Flag, 'packet_steering', _('Packet Steering'), _('Enable packet steering across all CPUs. May help or hinder network speed.'));
+		o.optional = true;
+
+
 		if (dslModemType != null) {
 			s = m.section(form.TypedSection, 'dsl', _('DSL'));
 			s.anonymous = true;