From 556dcc7a045f49e406f4ffd6b44e29357a82e131 Mon Sep 17 00:00:00 2001
From: garypang13 <48883331+garypang13@users.noreply.github.com>
Date: Mon, 14 Dec 2020 04:16:49 +0800
Subject: [PATCH] Update Config-kernel.in

---
 config/Config-kernel.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/config/Config-kernel.in b/config/Config-kernel.in
index 1cd9da4d12f1..a0b2f6210e15 100644
--- a/config/Config-kernel.in
+++ b/config/Config-kernel.in
@@ -44,6 +44,11 @@ config KERNEL_DEBUG_FS
 	  write to these files. Many common debugging facilities, such as
 	  ftrace, require the existence of debugfs.
 
+config KERNEL_MIPS_FPU_EMULATOR
+	bool "Compile the kernel with MIPS FPU Emulator"
+	default y
+	depends on (mips || mipsel || mips64 || mips64el)
+
 config KERNEL_MIPS_FP_SUPPORT
 	bool
 	default y if TARGET_pistachio
