--- a/package/feeds/custom/luci-app-passwall/luasrc/controller/passwall.lua
+++ b/package/feeds/custom/luci-app-passwall/luasrc/controller/passwall.lua
@@ -36,9 +36,11 @@ function index()
 	entry({"admin", "services", appname, "acl"}, cbi(appname .. "/client/acl"), _("Access control"), 98).leaf = true
 	entry({"admin", "services", appname, "log"}, form(appname .. "/client/log"), _("Watch Logs"), 999).leaf = true
 
+	if nixio.fs.access("/usr/bin/ssr-server") then
 	--[[ Server ]]
 	entry({"admin", "services", appname, "server"}, cbi(appname .. "/server/index"), _("Server-Side"), 99).leaf = true
 	entry({"admin", "services", appname, "server_user"}, cbi(appname .. "/server/user")).leaf = true
+	end
 
 	--[[ API ]]
 	entry({"admin", "services", appname, "server_user_status"}, call("server_user_status")).leaf = true

--- a/package/feeds/custom/luci-app-passwall/luasrc/model/cbi/passwall/client/auto_switch.lua
+++ b/package/feeds/custom/luci-app-passwall/luasrc/model/cbi/passwall/client/auto_switch.lua
@@ -37,11 +37,6 @@ o.default = "3"
 -- 暂时只支持TCP1
 local tcp_node_num = 1
 for i = 1, tcp_node_num, 1 do
-    o = s:option(ListValue, "tcp_main" .. i, "TCP " .. i .. " " .. translate("Main node"))
-    for k, v in pairs(nodes_table) do
-        o:value(v.id, v.remarks)
-    end
-    
     o = s:option(DynamicList, "tcp_node" .. i, "TCP " .. i .. " " .. translate("List of backup nodes"))
     for k, v in pairs(nodes_table) do
         o:value(v.id, v.remarks)

--- a/package/feeds/custom/luci-app-passwall/luasrc/view/passwall/rule/rule_version.htm
+++ b/package/feeds/custom/luci-app-passwall/luasrc/view/passwall/rule/rule_version.htm
@@ -26,13 +26,9 @@
 				update: checkBoxList
 			},
 			function(x, data) {
-				if(x && x.status == 200) {
-					window.location.href = '<%=url([[admin]], [[services]], [[passwall]], [[log]])%>';
-				} else {
 					alert("<%:Error%>");
 					btn.disabled = false;
 					btn.value = '<%:Manually update%>';
-				}
 			}
 		);
 	}

--- a/package/feeds/custom/luci-app-passwall/root/usr/share/passwall/app.sh
+++ b/package/feeds/custom/luci-app-passwall/root/usr/share/passwall/app.sh
@@ -933,8 +933,6 @@ add_dnsmasq() {
 		msg="ISP"
 		servers="${LOCAL_DNS}"
 		echo "conf-dir=${TMP_DNSMASQ_PATH}" > "/var/dnsmasq.d/dnsmasq-${CONFIG}.conf"
-		#兼容旧版dnsmasq
-		echo "conf-dir=${TMP_DNSMASQ_PATH}" > "${DNSMASQ_PATH}/dnsmasq-${CONFIG}.conf"
 
 		[ "${USE_CHNLIST}" = "1" ] && [ -z "${returnhome}" ] && [ -n "${chnlist}" ] && servers="${TUN_DNS}"
 		[ -n "${chnlist}" ] && msg="中国列表以外"