--- a/include/target.mk
+++ b/include/target.mk
@@ -53,15 +53,31 @@ DEFAULT_PACKAGES.nas:=\
 	mdadm
 # For router targets
 DEFAULT_PACKAGES.router:=\
-	dnsmasq \
-	firewall4 \
-	nftables \
-	kmod-nft-offload \
+	dnsmasq-full \
+	firewall \
+	kmod-ipt-nat \
+	kmod-ipt-na6 \
 	odhcp6c \
 	odhcpd-ipv6only \
 	ppp \
 	ppp-mod-pppoe
 
+ifneq ($(CONFIG_SMALL_FLASH),)
+DEFAULT_PACKAGES+=-coremark -htop -bash -openssh-sftp-server
+endif
+
+ifeq ($(CONFIG_ARCH),arm)
+DEFAULT_PACKAGES+=autocore-arm luci-app-cpufreq
+endif
+
+ifeq ($(CONFIG_ARCH),aarch64)
+DEFAULT_PACKAGES+=autocore-arm luci-app-cpufreq
+endif
+
+ifneq ($(CONFIG_USB_SUPPORT),)
+DEFAULT_PACKAGES+=automount
+endif
+
 ifneq ($(DUMP),)
   all: dumpinfo
 endif

--- a/target/Config.in
+++ b/target/Config.in
@@ -37,6 +37,7 @@ config PWM_SUPPORT
 
 config USB_SUPPORT
 	select AUDIO_SUPPORT
+	select PACKAGE_automount
 	bool
 
 config USB_GADGET_SUPPORT
@@ -123,6 +124,8 @@ config USES_BOOT_PART
 
 config aarch64
 	select ARCH_64BIT
+	select PACKAGE_autocore-arm
+	select PACKAGE_luci-app-cpufreq
 	bool
 
 config aarch64_be
@@ -138,6 +141,8 @@ config arceb
 	bool
 
 config arm
+	select PACKAGE_autocore-arm
+	select PACKAGE_luci-app-cpufreq
 	bool
 
 config armeb