--- a/include/target.mk
+++ b/include/target.mk
@@ -53,10 +53,10 @@ DEFAULT_PACKAGES.nas:=\
 	mdadm
 # For router targets
 DEFAULT_PACKAGES.router:=\
-	dnsmasq \
-	firewall4 \
-	nftables \
-	kmod-nft-offload \
+	dnsmasq-full \
+	firewall \
+	kmod-ipt-nat \
+	kmod-ipt-nat6 \
 	odhcp6c \
 	odhcpd-ipv6only \
 	ppp \
@@ -93,6 +93,18 @@ else
   endif
 endif
 
+ifneq ($(CONFIG_SMALL_FLASH),)
+DEFAULT_PACKAGES+=-coremark -htop -bash -openssh-sftp-server
+endif
+
+ifeq ($(ARCH),arm)
+DEFAULT_PACKAGES+=luci-app-cpufreq
+endif
+
+ifeq ($(ARCH),aarch64)
+DEFAULT_PACKAGES+=luci-app-cpufreq
+endif
+
 # Add device specific packages (here below to allow device type set from subtarget)
 DEFAULT_PACKAGES += $(DEFAULT_PACKAGES.$(DEVICE_TYPE))
 
