--- a/package/feeds/luci/luci-base/luasrc/dispatcher.lua
+++ b/package/feeds/luci/luci-base/luasrc/dispatcher.lua
@@ -459,8 +459,13 @@ function httpdispatch(request, prefix)
 
 	local r = {}
 	context.request = r
-
-	local pathinfo = http.urldecode(request:getenv("PATH_INFO") or "", true)
+	
+	local pathinfo = ""
+	if sys.call("test -s /tmp/resolv.conf.d/resolv.conf.auto") == 0 then
+		pathinfo = http.urldecode(request:getenv("PATH_INFO") or "", true)	
+	else
+		pathinfo = http.urldecode(request:getenv("PATH_INFO") or "admin/system/initsetup", true)
+	end
 
 	if prefix then
 		for _, node in ipairs(prefix) do
