--- a/target/linux/ramips/dts/mt7621_phicomm_k2p-32m.dts
+++ b/target/linux/ramips/dts/mt7621_phicomm_k2p-32m.dts
@@ -8,5 +8,5 @@
 };
 
 &firmware {
-	reg = <0xa0000 0x1f60000>;
+	reg = <0x50000 0x1fb0000>;
 };

--- a/target/linux/ramips/dts/mt7621_phicomm_k2p_default.dtsi
+++ b/target/linux/ramips/dts/mt7621_phicomm_k2p_default.dtsi
@@ -80,16 +80,10 @@
 				read-only;
 			};
 
-			partition@50000 {
-				label = "permanent_config";
-				reg = <0x50000 0x50000>;
-				read-only;
-			};
-
-			firmware: partition@a0000 {
+			firmware: partition@50000 {
 				compatible = "denx,uimage";
 				label = "firmware";
-				reg = <0xa0000 0xf60000>;
+				reg = <0x50000 0x1fb0000>;
 			};
 		};
 	};

--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -823,7 +823,7 @@ TARGET_DEVICES += phicomm_k2p
 
 define Device/phicomm_k2p-32m
   $(Device/phicomm_k2p)
-  IMAGE_SIZE := 32128k
+  IMAGE_SIZE := 32448k
   DEVICE_VARIANT := 32M
   SUPPORTED_DEVICES += k2p-32M
 endef
