--- a/include/image.mk
+++ b/include/image.mk
@@ -294,5 +294,5 @@
 endef
 
-ROOTFSTZ = $(IMG_PREFIX)$(if $(PROFILE_SANITIZED),-$(PROFILE_SANITIZED))-rootfs.tar.gz
+ROOTFSTZ = $(IMG_PREFIX)$(if $(PROFILE_SANITIZED),-$(PROFILE_SANITIZED)).tar.gz
 ifdef CONFIG_TARGET_ROOTFS_TARGZ
   define Image/Build/targz
@@ -308,6 +308,6 @@
 	FILE_NAME="$(ROOTFSTZ)" \
 	FILE_DIR="$(BIN_DIR)" \
-	FILE_TYPE="rootfs" \
-	FILE_FILESYSTEM="rootfs" \
+	FILE_TYPE="sysupgrade" \
+	FILE_FILESYSTEM="ext4" \
 	DEVICE_IMG_PREFIX="$(DEVICE_IMG_PREFIX)" \
 	DEVICE_VENDOR="$(shell echo $($(PROFILE)_NAME) | cut -d " " -f1)" \
