--- a/target/linux/rockchip/image/rk35xx.mk
+++ b/target/linux/rockchip/image/rk35xx.mk
@@ -78,6 +78,16 @@ $(call Device/rk3528)
 endef
 TARGET_DEVICES += radxa_e20c
 
+define Device/hlink_h28k
+$(call Device/rk3528)
+  DEVICE_VENDOR := Hlink
+  DEVICE_MODEL := H28K
+  SUPPORTED_DEVICES += hlink,h28k
+  DEVICE_DTS := rk3528-h28k
+  DEVICE_PACKAGES := kmod-r8169 kmod-r8125 kmod-thermal
+endef
+TARGET_DEVICES += hlink_h28k
+
 define Device/radxa_rock-5c
 $(call Device/rk3588)
   DEVICE_VENDOR := Radxa
