--- a/target/linux/x86/image/Makefile
+++ b/target/linux/x86/image/Makefile
@@ -108,22 +108,18 @@ define Device/Default
   IMAGE/combined.img := grub-config pc | combined | grub-install | append-metadata
   IMAGE/combined.img.gz := grub-config pc | combined | grub-install | gzip | append-metadata
   IMAGE/combined.vdi := grub-config pc | combined | grub-install | qemu-image vdi
-  IMAGE/combined.vmdk := grub-config pc | combined | grub-install | qemu-image vmdk
+  IMAGE/combined.vmdk.zip := grub-config pc | combined | grub-install | qemu-exsi vmdk -o adapter_type=lsilogic -o subformat=monolithicFlat
   IMAGE/combined.vhdx := grub-config pc | combined | grub-install | qemu-image vhdx -o subformat=dynamic
-  IMAGE/rootfs.img := append-rootfs | pad-to $(ROOTFS_PARTSIZE)
-  IMAGE/rootfs.img.gz := append-rootfs | pad-to $(ROOTFS_PARTSIZE) | gzip
   ARTIFACT/image-efi.iso := grub-config iso | iso efi
   IMAGE/combined-efi.img := grub-config efi | combined efi | grub-install efi | append-metadata
   IMAGE/combined-efi.img.gz := grub-config efi | combined efi | grub-install efi | gzip | append-metadata
   IMAGE/combined-efi.vdi := grub-config efi | combined efi | grub-install efi | qemu-image vdi
-  IMAGE/combined-efi.vmdk := grub-config efi | combined efi | grub-install efi | qemu-image vmdk
+  IMAGE/combined-efi.vmdk.zip := grub-config efi | combined efi | grub-install efi | qemu-exsi vmdk -o adapter_type=lsilogic -o subformat=monolithicFlat
   IMAGE/combined-efi.vhdx := grub-config efi | combined efi | grub-install efi | qemu-image vhdx -o subformat=dynamic
   ifeq ($(CONFIG_TARGET_IMAGES_GZIP),y)
-    IMAGES-y := rootfs.img.gz
     IMAGES-$$(CONFIG_GRUB_IMAGES) += combined.img.gz
     IMAGES-$$(CONFIG_GRUB_EFI_IMAGES) += combined-efi.img.gz
   else
-    IMAGES-y := rootfs.img
     IMAGES-$$(CONFIG_GRUB_IMAGES) += combined.img
     IMAGES-$$(CONFIG_GRUB_EFI_IMAGES) += combined-efi.img
   endif
@@ -139,8 +135,8 @@ define Device/Default
     IMAGES-$$(CONFIG_GRUB_EFI_IMAGES) += combined-efi.vdi
   endif
   ifeq ($(CONFIG_VMDK_IMAGES),y)
-    IMAGES-$$(CONFIG_GRUB_IMAGES) += combined.vmdk
-    IMAGES-$$(CONFIG_GRUB_EFI_IMAGES) += combined-efi.vmdk
+    IMAGES-$$(CONFIG_GRUB_IMAGES) += combined.vmdk.zip
+    IMAGES-$$(CONFIG_GRUB_EFI_IMAGES) += combined-efi.vmdk.zip
   endif
   ifeq ($(CONFIG_VHDX_IMAGES),y)
     IMAGES-$$(CONFIG_GRUB_IMAGES) += combined.vhdx
