--- a/package/boot/uboot-rockchip/Makefile
+++ b/package/boot/uboot-rockchip/Makefile
@@ -161,6 +161,19 @@ define U-Boot/nanopi-r6c-rk3588
 
 endef
 
+define U-Boot/orangepi-5-rk3588
+  BUILD_SUBTARGET:=armv8
+  NAME:=ORANGEPI-5
+  BUILD_DEVICES:= \
+    xunlong_orangepi-5
+  DEPENDS:=+PACKAGE_u-boot-orangepi-5-rk3588:arm-trusted-firmware-rockchip
+  PKG_BUILD_DEPENDS:=arm-trusted-firmware-rockchip
+  DEVICE_DTS := rk3588-nanopi-r6c.dts
+  UBOOT_CONFIG:= nanopi-r6c-rk3588
+  ATF:=rk3588_bl31.elf
+  DDR_3588:=$(1)
+endef
+
 define U-Boot/nanopi-r6s-rk3588
   BUILD_SUBTARGET:=armv8
   NAME:=NANOPI-R6C

--- a/target/linux/rockchip_rk3588/armv8/base-files/etc/board.d/02_network
+++ b/target/linux/rockchip_rk3588/armv8/base-files/etc/board.d/02_network
@@ -71,6 +71,7 @@ rockchip_setup_macs()
 	friendlyelec,nanopi-r6c|\
 	friendlyelec,nanopi-r6s|\
 	friendlyarm,nanopi-r4s|\
+	rockchip,rk3588s-orangepi-5|\
 	rockchip,rk3588-orangepi-5-plus)
 		wan_mac=$(nanopi_r4s_get_mac wan)
 		lan_mac=$(nanopi_r4s_get_mac lan)

--- a/target/linux/rockchip_rk3588/image/armv8.mk
+++ b/target/linux/rockchip_rk3588/image/armv8.mk
@@ -53,6 +53,23 @@ define Device/friendlyelec_nanopi-r6s
 endef
 TARGET_DEVICES += friendlyelec_nanopi-r6s
 
+define Device/xunlong_orangepi-5
+  DEVICE_VENDOR := XunLong
+  DEVICE_MODEL := Orange Pi 5
+  SOC := rk3588s
+  UBOOT_DEVICE_NAME := orangepi-5-rk3588
+  SUPPORTED_DEVICES := rockchip,rk3588s-orangepi-5
+  IMAGE/sysupgrade.img.gz := boot-common | boot-script nanopi-r6c | pine64-img | gzip | append-metadata
+ DEVICE_PACKAGES := \
+	blkid block-mount blockd btrfs-progs f2fs-tools f2fsck fdisk fixparts fstools gdisk \
+	kmod-ata-ahci kmod-ata-core kmod-fs-autofs4 kmod-fs-btrfs kmod-fs-exfat kmod-fs-exportfs luci-app-ttyd \
+	kmod-fs-ext4 kmod-fs-squashfs kmod-nvme kmod-usb-net-rtl8152 kmod-usb-storage kmod-usb-storage-extras kmod-usb-storage-uas \
+	kmod-usb-xhci-hcd kmod-usb3 losetup lsblk luci luci-proto-wireguard mkf2fs mount-utils parted partx-util pciutils resize2fs \
+	kmod-r8125 usbutils wget-ssl smartmontools luci-app-statistics collectd-mod-cpufreq collectd-mod-sensors collectd-mod-thermal \
+	collectd-mod-conntrack collectd-mod-irq dnsmasq-full -dnsmasq ethtool iperf3-ssl bash irqbalance
+endef
+TARGET_DEVICES += xunlong_orangepi-5
+
 define Device/xunlong_orangepi-5-plus
   DEVICE_VENDOR := XunLong
   DEVICE_MODEL := OrangePi 5 Plus

