--- a/target/linux/amlogic/image/Makefile
+++ b/target/linux/amlogic/image/Makefile
@@ -36,7 +36,7 @@ endef
 ### Devices ###
 define Device/Default
   FILESYSTEMS := ext4
-  IMAGES := emmc.img
+  IMAGES := emmc_burn.img
   KERNEL_DEPENDS = $$(wildcard $(DTS_DIR)/$$(DEVICE_DTS).dts)
   KERNEL_LOADADDR := 0x01080000
   KERNEL_NAME := Image
@@ -46,9 +46,10 @@ endef
 
 define Device/thunder-onecloud
   DEVICE_DTS := amlogic/meson8b-onecloud
-  DEVICE_TITLE := Thunder OneCloud
+  DEVICE_TITLE := OneCloud 玩客云
+  DEVICE_PACKAGES += kmod-usb-net-rtl8152
   KERNEL_LOADADDR := 0x00208000
-  IMAGE/emmc.img := boot-script onecloud | emmc-common $$(DEVICE_NAME)
+  IMAGE/emmc_burn.img := boot-script onecloud | emmc-common $$(DEVICE_NAME)
 endef
 ifeq ($(SUBTARGET),meson8b)
   TARGET_DEVICES += thunder-onecloud
