From fe7d7074c4dc75eaf868addc4d4c602e4233e31f Mon Sep 17 00:00:00 2001
From: kiddin9 <48883331+kiddin9@users.noreply.github.com>
Date: Sat, 23 Apr 2022 22:14:29 +0800
Subject: [PATCH] Update modules.mk

---
 target/linux/rockchip/modules.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/target/linux/rockchip/modules.mk b/target/linux/rockchip/modules.mk
index 4ad49b57e82..46f3e715624 100644
--- a/target/linux/rockchip/modules.mk
+++ b/target/linux/rockchip/modules.mk
@@ -35,7 +35,8 @@ define KernelPackage/drm-rockchip
 	$(LINUX_DIR)/drivers/media/cec/core/cec.ko@ge5.10 \
 	$(LINUX_DIR)/drivers/phy/rockchip/phy-rockchip-inno-hdmi.ko \
 	$(LINUX_DIR)/drivers/gpu/drm/panel/panel-simple.ko \
-	$(LINUX_DIR)/drivers/gpu/drm/rockchip/rockchipdrm.ko
+	$(LINUX_DIR)/drivers/gpu/drm/rockchip/rockchipdrm.ko \
+	$(LINUX_DIR)/drivers/gpu/drm/drm_dp_aux_bus.ko@gt5.10
   AUTOLOAD:=$(call AutoProbe,rockchipdrm phy-rockchip-inno-hdmi dw-hdmi-cec)
 endef
 
