--- a/target/linux/armvirt_pcat/Makefile
+++ b/target/linux/armvirt_pcat/Makefile
@@ -4,15 +4,15 @@
 
 include $(TOPDIR)/rules.mk
 
-BOARD:=armvirt
+BOARD:=armvirt_pcat
 BOARDNAME:=QEMU ARM Virtual Machine
 FEATURES:=fpu pci rtc usb
-FEATURES+=cpiogz ext4 ramdisk squashfs targz
+FEATURES+=cpiogz ext4 ramdisk squashfs targz boot-part rootfs-part
 
 KERNEL_PATCHVER:=5.15
 
 include $(INCLUDE_DIR)/target.mk
 
-DEFAULT_PACKAGES += mkf2fs e2fsprogs
+DEFAULT_PACKAGES += mkf2fs e2fsprogs blkid pcat-manager -luci-app-gpsysupgrade
 
 $(eval $(call BuildTarget))