--- a/src/dnsmasq.h
+++ b/src/dnsmasq.h
@@ -271,3 +271,4 @@
 #define OPT_LEASE_RENEW    61
-#define OPT_LAST           62
+#define OPT_FILTER_AAAA    62
+#define OPT_LAST           63
 

--- a/src/option.c
+++ b/src/option.c
@@ -169,2 +169,3 @@
 #define LOPT_SCRIPT_TIME   360
+#define LOPT_FILTER_AAAA   361
  
@@ -341,2 +342,3 @@
     { "dhcp-ignore-clid", 0, 0,  LOPT_IGNORE_CLID },
+    { "filter-aaaa", 0, 0, LOPT_FILTER_AAAA },
     { NULL, 0, 0, 0 }
@@ -520,2 +522,3 @@
   { LOPT_SCRIPT_TIME, OPT_LEASE_RENEW, NULL, gettext_noop("Call dhcp-script when lease expiry changes."), NULL },
+  { LOPT_FILTER_AAAA, OPT_FILTER_AAAA, NULL, gettext_noop("Filter all AAAA requests."), NULL },
   { 0, 0, NULL, NULL, NULL }

--- a/src/rfc1035.c
+++ b/src/rfc1035.c
@@ -1957,2 +1957,9 @@
 
+      if (qtype == T_AAAA && option_bool(OPT_FILTER_AAAA) ){
+        //return a null reply
+        ans = 1;
+        if (!dryrun) log_query(F_CONFIG | F_IPV6 | F_NEG, name, &addr, NULL);
+        break;
+      }
+
       if (!ans)
