--- a/target/linux/ipq60xx/Makefile
+++ b/target/linux/ipq60xx/Makefile
@@ -16,7 +16,7 @@ include $(INCLUDE_DIR)/target.mk
 DEFAULT_PACKAGES += \
 	kmod-leds-gpio kmod-gpio-button-hotplug swconfig \
 	kmod-qca-nss-dp kmod-qca-nss-drv kmod-qca-ssdk \
-	kmod-ath11k-ahb qca-ssdk-shell wpad-openssl \
+	kmod-ath11k-ahb qca-ssdk-shell wpad-basic-wolfssl \
 	ath11k-firmware-ipq6018 nss-firmware-ipq6018 \
 	kmod-qca-nss-drv-pppoe kmod-qca-nss-ecm uboot-envtools
 

--- a/target/linux/ipq60xx/image/Makefile
+++ b/target/linux/ipq60xx/image/Makefile
@@ -104,7 +104,7 @@ define Device/qihoo_v6
 	PAGESIZE := 2048
 	DEVICE_DTS := qcom-ipq6018-qihoo-v6
 	DEVICE_DTS_CONFIG := config@cp03-c1
-	DEVICE_TITLE := Qihoo V6
+	DEVICE_TITLE := Qihoo 360 V6
 	DEVICE_PACKAGES := ath11k-wifi-qihoo-v6 kmod-usb3 kmod-usb-phy-msm
 endef
 TARGET_DEVICES += qihoo_v6

--- a/scripts/json_overview_image_info.py
+++ b/scripts/json_overview_image_info.py
@@ -33,7 +33,7 @@
             )
 
 if output:
-    default_packages, output["arch_packages"] = run(
+    default_packages, output["arch_packages"], output["kernel_version"], output["initramfs"] = run(
         [
             "make",
             "--no-print-directory",
@@ -41,6 +41,8 @@
             "target/linux/{}".format(output['target'].split('/')[0]),
             "val.DEFAULT_PACKAGES",
             "val.ARCH_PACKAGES",
+            "val.LINUX_VERSION",
+            "val.KERNEL_INITRAMFS",
         ],
         stdout=PIPE,
         stderr=PIPE,

--- a/rules.mk
+++ b/rules.mk
@@ -62,6 +62,7 @@ _SINGLE=export MAKEFLAGS=$(space);
 CFLAGS:=
 ARCH:=$(subst i486,i386,$(subst i586,i386,$(subst i686,i386,$(call qstrip,$(CONFIG_ARCH)))))
 ARCH_PACKAGES:=$(call qstrip,$(CONFIG_TARGET_ARCH_PACKAGES))
+KERNEL_INITRAMFS:=$(call qstrip,$(CONFIG_TARGET_ROOTFS_INITRAMFS))
 BOARD:=$(call qstrip,$(CONFIG_TARGET_BOARD))
 SUBTARGET:=$(call qstrip,$(CONFIG_TARGET_SUBTARGET))
 TARGET_OPTIMIZATION:=$(call qstrip,$(CONFIG_TARGET_OPTIMIZATION))