--- a/target/linux/armsr/image/Makefile
+++ b/target/linux/armsr/image/Makefile
@@ -81,10 +81,8 @@ define Device/efi-default
   IMAGE/combined.img.gz := grub-config efi | combined efi | grub-install efi | gzip | append-metadata
   IMAGE/combined.vmdk := grub-config efi | combined efi | grub-install efi | qemu-image vmdk
  ifeq ($(CONFIG_TARGET_IMAGES_GZIP),y)
-    IMAGES-y := rootfs.img.gz
     IMAGES-y += combined.img.gz
   else
-    IMAGES-y := rootfs.img
     IMAGES-y += combined.img
   endif
   ifeq ($(CONFIG_VMDK_IMAGES),y)
@@ -116,4 +114,24 @@ define Device/generic
 endef
 TARGET_DEVICES += generic
 
+define Device/box
+  DEVICE_TITLE := 电视盒子全系列 / N1
+  DEVICE_PACKAGES := perlbase-base perlbase-utf8 perlbase-time perlbase-xsloader perlbase-file btrfs-progs luci-app-amlogic kmod-brcmfmac wpad-basic-mbedtls iw fdisk lsblk automount
+  IMAGE/rootfs.tar.gz := append-rootfs | gzip
+  IMAGES := rootfs.tar.gz
+  KERNEL := kernel-bin
+endef
+TARGET_DEVICES += box
+
+define Image/Build
+  if [ "$(PROFILE_SANITIZED)" == "box" ]; then \
+  	export BIN_DIR=$(BIN_DIR); \
+  	export DATE=$(DATE); \
+  	export MORE="$(MORE)"; \
+	export ROOTFS_PARTSIZE=$(shell echo $$(($(ROOTFS_PARTSIZE)/1024/1024))); \
+  	cd /data/packit/amlogic-s9xxx-openwrt; \
+  	. ~/packit/packit_amlogic.sh; \
+  fi
+endef
+
 $(eval $(call BuildImage))
